{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Directorio con la carpeta donde se encuentran los datos. Cambiar si fuera necesario.\n",
    "work_folder = '/Directorio/SL_Data/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Genes esenciales (Selección negativa) y supresores (Selección positiva) inferidos del análisis de MAGeCKFlute.\n",
    "\n",
    "# PatuT. Genes esenciales:\n",
    "PatuTEss = ['PRMT3','ASXL1','HELLS','SMARCE1','KDM4B','EPC2']\n",
    "# PatuT. Genes supresores:\n",
    "PatuTSup = ['WHSC1','SUV39H2']\n",
    "\n",
    "# PatuS. Genes esenciales:\n",
    "PatuSEss = ['CHD7','BRD4','ING5']\n",
    "# PatuS. Genes supresores:\n",
    "PatuSSup = ['PHF23','UBE2A']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Función que comprueba si de dentre los genes de una lista de parejas de SL se encuentran los introducidos, y devuelve la lista de parejas completa.\n",
    "def SLfinder(df, gene):\n",
    "    n2 = df.loc[df['n1.name'] == gene, 'n2.name']\n",
    "    n1 = df.loc[df['n2.name'] == gene, 'n1.name']\n",
    "    n = list(n2) + list(n1)\n",
    "    if len(n) > 0:\n",
    "        return n\n",
    "    else:\n",
    "        return 'No SL pairs Found'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Función que aplica la función SLfiner() a la lista de genes definidos anteriormente \n",
    "# usando los datos introducidos de una base de datos concreta. Después guarda las parejas\n",
    "# encontradas por cada gen en un archivo .csv.\n",
    "def SLcsv(df,db):\n",
    "    SL_results = {}\n",
    "\n",
    "    print(f'-------SL pairs for Essential PatuT Genes in {db}:')\n",
    "    print()\n",
    "    for gene in PatuTEss:\n",
    "        SLpairsTEssAll = SLfinder(df, gene)\n",
    "        print(f'SL pairs in {db} for {gene}: {SLpairsTEssAll}')\n",
    "        if isinstance(SLpairsTEssAll, list):\n",
    "            SL_results[gene] = SLpairsTEssAll\n",
    "        print()\n",
    "    \n",
    "    print(f'-------SL pairs for Suppressor PatuT Genes in {db}:')\n",
    "    print()\n",
    "    for gene in PatuTSup:\n",
    "        SLpairsTSupAll = SLfinder(df, gene)\n",
    "        print(f'SL pairs in {db} for {gene}: {SLpairsTSupAll}')\n",
    "        if isinstance(SLpairsTSupAll, list):\n",
    "            SL_results[gene] = SLpairsTSupAll\n",
    "        print()\n",
    "\n",
    "    print(f'-------SL pairs for Essential PatuS Genes in {db}:')\n",
    "    print()\n",
    "    for gene in PatuSEss:\n",
    "        SLpairsSEssAll = SLfinder(df, gene)\n",
    "        print(f'SL pairs in {db} for {gene}: {SLpairsSEssAll}')\n",
    "        if isinstance(SLpairsSEssAll, list):\n",
    "            SL_results[gene] = SLpairsSEssAll\n",
    "        print()\n",
    "\n",
    "    print(f'-------SL pairs for Suppressor PatuS Genes in {db}:')\n",
    "    print()\n",
    "    for gene in PatuSSup:\n",
    "        SLpairsSSupAll = SLfinder(df, gene)\n",
    "        print(f'SL pairs in {db} for {gene}: {SLpairsSSupAll}')\n",
    "        if isinstance(SLpairsSSupAll, list):\n",
    "            SL_results[gene] = SLpairsSSupAll\n",
    "        print()\n",
    "\n",
    "    if len(SL_results) != 0:\n",
    "\n",
    "        SL_results = pd.DataFrame.from_dict(SL_results, orient = 'index').T\n",
    "        SL_results = SL_results.map(lambda x: '' if x is None else x)\n",
    "        SL_results.to_csv(f'{work_folder}Candidate_SL_Pairs/{db}_SLPairs.csv', header = True, index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# SynLethDB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       n1.name  n1.identifier  n2.name  n2.identifier r.cell_line r.pubmed_id  \\\n",
      "0      TMEM35A          59353     PTEN           5728         NaN    26427375   \n",
      "1      TMEM35A          59353    BRCA2            675         NaN    26427375   \n",
      "2        CIDEA           1149     NAE1           8883        A375    23100467   \n",
      "3         FHL5           9457    OPRK1           4986         NaN    25171417   \n",
      "4         FHL5           9457    GHRHR           2692         NaN    25171417   \n",
      "...        ...            ...      ...            ...         ...         ...   \n",
      "35938    PRKCQ           5588  SLC22A1           6580   K562;K562    20609354   \n",
      "35939   RNF19A          25897     TP53           7157         NaN    26427375   \n",
      "35940  SLC10A7          84068      ATM            472         NaN    26427375   \n",
      "35941     SVIL           6840     PTEN           5728         NaN    26427375   \n",
      "35942     SVIL           6840     CAV2            858         NaN    25171417   \n",
      "\n",
      "                       r.source  r.statistic_score  \n",
      "0      Computational Prediction                0.3  \n",
      "1      Computational Prediction                0.3  \n",
      "2                    GenomeRNAi                0.5  \n",
      "3      Computational Prediction                0.3  \n",
      "4      Computational Prediction                0.3  \n",
      "...                         ...                ...  \n",
      "35938                GenomeRNAi              521.0  \n",
      "35939  Computational Prediction                0.3  \n",
      "35940  Computational Prediction                0.3  \n",
      "35941  Computational Prediction                0.3  \n",
      "35942  Computational Prediction                0.3  \n",
      "\n",
      "[35943 rows x 8 columns]\n"
     ]
    }
   ],
   "source": [
    "# Carga de los datos de SynLethDB.\n",
    "SLdb = pd.read_csv(f'{work_folder}SynLethDB/Human_SL.csv')\n",
    "print(SLdb)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Text Mining\n",
      "CRISPR/CRISPRi\n",
      "GenomeRNAi\n",
      "Decipher\n",
      "High Throughput|Low Throughput\n",
      "Low Throughput\n",
      "Synlethality\n",
      "Drug Screen\n",
      "High Throughput\n",
      "Computational Prediction\n",
      "Daisy\n",
      "RNAi Screen\n"
     ]
    }
   ],
   "source": [
    "# Obtención de una lista con todos los posibles orígenes de cada pareja registarda en SynLethDB.\n",
    "SLdbSource = SLdb['r.source'].astype('category')\n",
    "Source_list = SLdbSource.cat.categories.tolist()\n",
    "\n",
    "Source_types = set()\n",
    "\n",
    "# División de aquellas entradas con dos o más fuentes para obtenerlas por separado.\n",
    "for source in Source_list:\n",
    "    \n",
    "    sources = re.split(r'[;,]', source)\n",
    "    \n",
    "    Source_types.update(s.strip() for s in sources)\n",
    "\n",
    "for s in Source_types:\n",
    "    print(s)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       n1.name  n1.identifier  n2.name  n2.identifier r.cell_line r.pubmed_id  \\\n",
      "0      TMEM35A          59353     PTEN           5728         NaN    26427375   \n",
      "1      TMEM35A          59353    BRCA2            675         NaN    26427375   \n",
      "3         FHL5           9457    OPRK1           4986         NaN    25171417   \n",
      "4         FHL5           9457    GHRHR           2692         NaN    25171417   \n",
      "5         FHL5           9457  SLC13A1           6561         NaN    25171417   \n",
      "...        ...            ...      ...            ...         ...         ...   \n",
      "35934    NUP88           4927    DHX36         170506         NaN    31340155   \n",
      "35939   RNF19A          25897     TP53           7157         NaN    26427375   \n",
      "35940  SLC10A7          84068      ATM            472         NaN    26427375   \n",
      "35941     SVIL           6840     PTEN           5728         NaN    26427375   \n",
      "35942     SVIL           6840     CAV2            858         NaN    25171417   \n",
      "\n",
      "                       r.source  r.statistic_score  \n",
      "0      Computational Prediction             0.3000  \n",
      "1      Computational Prediction             0.3000  \n",
      "3      Computational Prediction             0.3000  \n",
      "4      Computational Prediction             0.3000  \n",
      "5      Computational Prediction             0.3000  \n",
      "...                         ...                ...  \n",
      "35934  Computational Prediction             0.1227  \n",
      "35939  Computational Prediction             0.3000  \n",
      "35940  Computational Prediction             0.3000  \n",
      "35941  Computational Prediction             0.3000  \n",
      "35942  Computational Prediction             0.3000  \n",
      "\n",
      "[9331 rows x 8 columns]\n",
      "0        Computational Prediction\n",
      "1        Computational Prediction\n",
      "3        Computational Prediction\n",
      "4        Computational Prediction\n",
      "5        Computational Prediction\n",
      "                   ...           \n",
      "35934    Computational Prediction\n",
      "35939    Computational Prediction\n",
      "35940    Computational Prediction\n",
      "35941    Computational Prediction\n",
      "35942    Computational Prediction\n",
      "Name: r.source, Length: 9331, dtype: category\n",
      "Categories (3, object): ['Computational Prediction', 'Decipher;Computational Prediction', 'Text Mining;Daisy']\n",
      "      n1.name  n1.identifier  n2.name  n2.identifier r.cell_line r.pubmed_id  \\\n",
      "2       CIDEA           1149     NAE1           8883        A375    23100467   \n",
      "6      FKBP11          51303    NTRK1           4914   K562;K562    20609354   \n",
      "7      FKBP11          51303      RET           5979   K562;K562    20609354   \n",
      "8      FKBP11          51303   PDGFRA           5156   K562;K562    20609354   \n",
      "9      FKBP11          51303  SLC22A1           6580   K562;K562    20609354   \n",
      "...       ...            ...      ...            ...         ...         ...   \n",
      "35933   CEP63          80254    EIF3C           8663        K562    31300006   \n",
      "35935   PRKCQ           5588    PTGS1           5742   K562;K562    20609354   \n",
      "35936   PRKCQ           5588  SLC22A2           6582   K562;K562    20609354   \n",
      "35937   PRKCQ           5588      RET           5979   K562;K562    20609354   \n",
      "35938   PRKCQ           5588  SLC22A1           6580   K562;K562    20609354   \n",
      "\n",
      "             r.source  r.statistic_score  \n",
      "2          GenomeRNAi               0.50  \n",
      "6          GenomeRNAi             506.00  \n",
      "7          GenomeRNAi             506.00  \n",
      "8          GenomeRNAi             506.00  \n",
      "9          GenomeRNAi             506.00  \n",
      "...               ...                ...  \n",
      "35933  CRISPR/CRISPRi               0.85  \n",
      "35935      GenomeRNAi             521.00  \n",
      "35936      GenomeRNAi             521.00  \n",
      "35937      GenomeRNAi             521.00  \n",
      "35938      GenomeRNAi             521.00  \n",
      "\n",
      "[26612 rows x 8 columns]\n",
      "2            GenomeRNAi\n",
      "6            GenomeRNAi\n",
      "7            GenomeRNAi\n",
      "8            GenomeRNAi\n",
      "9            GenomeRNAi\n",
      "              ...      \n",
      "35933    CRISPR/CRISPRi\n",
      "35935        GenomeRNAi\n",
      "35936        GenomeRNAi\n",
      "35937        GenomeRNAi\n",
      "35938        GenomeRNAi\n",
      "Name: r.source, Length: 26612, dtype: category\n",
      "Categories (17, object): ['CRISPR/CRISPRi', 'Decipher', 'Decipher;Text Mining', 'Drug Screen', ..., 'Synlethality;GenomeRNAi', 'Synlethality;Text Mining', 'Text Mining', 'Text Mining;Synlethality']\n"
     ]
    }
   ],
   "source": [
    "# Clasificación de las fuentes computacionales:\n",
    "CompSources = ['Computational Prediction','Decipher;Computational Prediction','Text Mining;Daisy']\n",
    "\n",
    "# Clasificación de las fuentes bibliográficas y/o experimentales:\n",
    "ExpSources = ['CRISPR/CRISPRi','Decipher', 'Decipher;Text Mining', 'Drug Screen', 'GenomeRNAi', 'GenomeRNAi;Decipher', 'GenomeRNAi;Text Mining', 'High Throughput', \n",
    "              'High Throughput|Low Throughput', 'Low Throughput', 'RNAi Screen', 'Synlethality', 'Synlethality;Decipher', 'Synlethality;GenomeRNAi', 'Synlethality;Text Mining', \n",
    "              'Text Mining','Text Mining;Synlethality']\n",
    "\n",
    "# Filtrado de los datos de SynLethDB según el origen de sus datos.\n",
    "# Fuentes computacionales:\n",
    "SLdbComp = SLdb[SLdb['r.source'].isin(CompSources)]\n",
    "print(SLdbComp)\n",
    "print(SLdbComp['r.source'].astype('category'))\n",
    "# Fuentes bibliográficas/experimentales:\n",
    "SLdbExp = SLdb[SLdb['r.source'].isin(ExpSources)]\n",
    "print(SLdbExp)\n",
    "print(SLdbExp['r.source'].astype('category'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------SL pairs for Essential PatuT Genes in SynLethDB_Comp:\n",
      "\n",
      "SL pairs in SynLethDB_Comp for PRMT3: No SL pairs Found\n",
      "\n",
      "SL pairs in SynLethDB_Comp for ASXL1: ['BRCA1']\n",
      "\n",
      "SL pairs in SynLethDB_Comp for HELLS: ['RFC2', 'TOPBP1', 'PCNA', 'ZNF107', 'SFPQ', 'FEN1']\n",
      "\n",
      "SL pairs in SynLethDB_Comp for SMARCE1: ['LRRC37A3']\n",
      "\n",
      "SL pairs in SynLethDB_Comp for KDM4B: ['SMARCC2', 'MED16', 'POGZ', 'DAZAP1']\n",
      "\n",
      "SL pairs in SynLethDB_Comp for EPC2: No SL pairs Found\n",
      "\n",
      "-------SL pairs for Suppressor PatuT Genes in SynLethDB_Comp:\n",
      "\n",
      "SL pairs in SynLethDB_Comp for WHSC1: No SL pairs Found\n",
      "\n",
      "SL pairs in SynLethDB_Comp for SUV39H2: ['ZWINT', 'RAD51', 'RBM17', 'KIF11', 'GTPBP4', 'ANKRD16', 'ATP5F1C']\n",
      "\n",
      "-------SL pairs for Essential PatuS Genes in SynLethDB_Comp:\n",
      "\n",
      "SL pairs in SynLethDB_Comp for CHD7: ['KCNJ8', 'HBG1', 'PRTN3']\n",
      "\n",
      "SL pairs in SynLethDB_Comp for BRD4: ['PRKCSH', 'GGA3', 'CCNT1', 'SIN3B', 'TP53', 'TNK2']\n",
      "\n",
      "SL pairs in SynLethDB_Comp for ING5: ['MRPS25']\n",
      "\n",
      "-------SL pairs for Suppressor PatuS Genes in SynLethDB_Comp:\n",
      "\n",
      "SL pairs in SynLethDB_Comp for PHF23: ['GPS2']\n",
      "\n",
      "SL pairs in SynLethDB_Comp for UBE2A: ['WDR44']\n",
      "\n"
     ]
    }
   ],
   "source": [
    "SLcsv(SLdbComp,'SynLethDB_Comp')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------SL pairs for Essential PatuT Genes in SynLethDB_Exp:\n",
      "\n",
      "SL pairs in SynLethDB_Exp for PRMT3: No SL pairs Found\n",
      "\n",
      "SL pairs in SynLethDB_Exp for ASXL1: ['MUS81']\n",
      "\n",
      "SL pairs in SynLethDB_Exp for HELLS: No SL pairs Found\n",
      "\n",
      "SL pairs in SynLethDB_Exp for SMARCE1: No SL pairs Found\n",
      "\n",
      "SL pairs in SynLethDB_Exp for KDM4B: No SL pairs Found\n",
      "\n",
      "SL pairs in SynLethDB_Exp for EPC2: No SL pairs Found\n",
      "\n",
      "-------SL pairs for Suppressor PatuT Genes in SynLethDB_Exp:\n",
      "\n",
      "SL pairs in SynLethDB_Exp for WHSC1: No SL pairs Found\n",
      "\n",
      "SL pairs in SynLethDB_Exp for SUV39H2: ['KRAS', 'NRAS', 'HRAS']\n",
      "\n",
      "-------SL pairs for Essential PatuS Genes in SynLethDB_Exp:\n",
      "\n",
      "SL pairs in SynLethDB_Exp for CHD7: ['KRAS', 'NRAS', 'HRAS']\n",
      "\n",
      "SL pairs in SynLethDB_Exp for BRD4: ['MYC', 'PTAR1', 'PARP1', 'NAE1', 'KMT2D', 'MSH6', 'KDM5C', 'PTEN', 'MAP2K1', 'FGFR3', 'RB1', 'DHFR', 'CHEK2', 'CDC7', 'BRCA1', 'CDK9', 'CHEK1', 'WEE1']\n",
      "\n",
      "SL pairs in SynLethDB_Exp for ING5: ['CDC7', 'TUBA1A', 'MAP2K1', 'CDK9']\n",
      "\n",
      "-------SL pairs for Suppressor PatuS Genes in SynLethDB_Exp:\n",
      "\n",
      "SL pairs in SynLethDB_Exp for PHF23: No SL pairs Found\n",
      "\n",
      "SL pairs in SynLethDB_Exp for UBE2A: ['NAE1']\n",
      "\n"
     ]
    }
   ],
   "source": [
    "SLcsv(SLdbExp,'SynLethDB_Exp')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Slorth"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "               0       1                2                3           4  \\\n",
      "0          CHEK1    MTOR  ENSG00000149554  ENSG00000198793  H. sapiens   \n",
      "1            ADA   CHEK1  ENSG00000196839  ENSG00000149554  H. sapiens   \n",
      "2          BRCA1   PARP1  ENSG00000012048  ENSG00000143799  H. sapiens   \n",
      "3          BRCA2   PARP1  ENSG00000139618  ENSG00000143799  H. sapiens   \n",
      "4       C10orf76   PTAR1  ENSG00000120029  ENSG00000188647  H. sapiens   \n",
      "...          ...     ...              ...              ...         ...   \n",
      "518631       TPR   U2AF1  ENSG00000047410  ENSG00000160201  H. sapiens   \n",
      "518632       TPR   VTI1A  ENSG00000047410  ENSG00000151532  H. sapiens   \n",
      "518633      TSC2   U2AF1  ENSG00000103197  ENSG00000160201  H. sapiens   \n",
      "518634      TSC2   VTI1A  ENSG00000103197  ENSG00000151532  H. sapiens   \n",
      "518635      TSC2  ZBTB16  ENSG00000103197  ENSG00000109906  H. sapiens   \n",
      "\n",
      "              5                          6         7  \n",
      "0       BioGRID  2208318, 2342099, 2342170  3.000000  \n",
      "1       BioGRID           2208350, 2342104  2.000000  \n",
      "2       BioGRID           2208315, 2342182  2.000000  \n",
      "3       BioGRID           2342181, 2344243  2.000000  \n",
      "4       BioGRID           2197822, 2197871  2.000000  \n",
      "...         ...                        ...       ...  \n",
      "518631   Slorth                          -  0.822218  \n",
      "518632   Slorth                          -  0.814406  \n",
      "518633   Slorth                          -  0.816285  \n",
      "518634   Slorth                          -  0.855900  \n",
      "518635   Slorth                          -  0.743188  \n",
      "\n",
      "[518636 rows x 8 columns]\n"
     ]
    }
   ],
   "source": [
    "# Carga de los datos de Slorth:\n",
    "SLslorth = pd.read_csv(f'Slorth/h.sapiens_ssl_predictions.csv', sep = '\\t', header = None)\n",
    "print(SLslorth)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "            0      1                2                3           4        5  \\\n",
      "0       CHEK1   MTOR  ENSG00000149554  ENSG00000198793  H. sapiens  BioGRID   \n",
      "1         ADA  CHEK1  ENSG00000196839  ENSG00000149554  H. sapiens  BioGRID   \n",
      "2       BRCA1  PARP1  ENSG00000012048  ENSG00000143799  H. sapiens  BioGRID   \n",
      "3       BRCA2  PARP1  ENSG00000139618  ENSG00000143799  H. sapiens  BioGRID   \n",
      "4    C10orf76  PTAR1  ENSG00000120029  ENSG00000188647  H. sapiens  BioGRID   \n",
      "..        ...    ...              ...              ...         ...      ...   \n",
      "928    TUBA1A  YWHAZ  ENSG00000167552  ENSG00000164924  H. sapiens  BioGRID   \n",
      "929     VEGFA    VHL  ENSG00000112715  ENSG00000134086  H. sapiens  BioGRID   \n",
      "930      WEE1   WEE1  ENSG00000166483  ENSG00000166483  H. sapiens  BioGRID   \n",
      "931      WEE1    WRN  ENSG00000166483  ENSG00000165392  H. sapiens  BioGRID   \n",
      "932      WEE1  XRCC3  ENSG00000166483  ENSG00000126215  H. sapiens  BioGRID   \n",
      "\n",
      "                             6    7  \n",
      "0    2208318, 2342099, 2342170  3.0  \n",
      "1             2208350, 2342104  2.0  \n",
      "2             2208315, 2342182  2.0  \n",
      "3             2342181, 2344243  2.0  \n",
      "4             2197822, 2197871  2.0  \n",
      "..                         ...  ...  \n",
      "928                    2342226  1.0  \n",
      "929                    2208434  1.0  \n",
      "930                    2342164  1.0  \n",
      "931                    2342159  1.0  \n",
      "932                    2342166  1.0  \n",
      "\n",
      "[933 rows x 8 columns]\n",
      "            0       1                2                3           4       5  \\\n",
      "933     CDC27   MYLIP  ENSG00000004897  ENSG00000007944  H. sapiens  Slorth   \n",
      "934     CDC27  NCAPD2  ENSG00000004897  ENSG00000010292  H. sapiens  Slorth   \n",
      "935     CDC27  POLR3B  ENSG00000004897  ENSG00000013503  H. sapiens  Slorth   \n",
      "936     CDC27    CDH1  ENSG00000004897  ENSG00000039068  H. sapiens  Slorth   \n",
      "937     CDC27   VAMP3  ENSG00000004897  ENSG00000049245  H. sapiens  Slorth   \n",
      "...       ...     ...              ...              ...         ...     ...   \n",
      "518631    TPR   U2AF1  ENSG00000047410  ENSG00000160201  H. sapiens  Slorth   \n",
      "518632    TPR   VTI1A  ENSG00000047410  ENSG00000151532  H. sapiens  Slorth   \n",
      "518633   TSC2   U2AF1  ENSG00000103197  ENSG00000160201  H. sapiens  Slorth   \n",
      "518634   TSC2   VTI1A  ENSG00000103197  ENSG00000151532  H. sapiens  Slorth   \n",
      "518635   TSC2  ZBTB16  ENSG00000103197  ENSG00000109906  H. sapiens  Slorth   \n",
      "\n",
      "        6         7  \n",
      "933     -  0.735260  \n",
      "934     -  0.853135  \n",
      "935     -  0.861576  \n",
      "936     -  0.783209  \n",
      "937     -  0.706373  \n",
      "...    ..       ...  \n",
      "518631  -  0.822218  \n",
      "518632  -  0.814406  \n",
      "518633  -  0.816285  \n",
      "518634  -  0.855900  \n",
      "518635  -  0.743188  \n",
      "\n",
      "[517703 rows x 8 columns]\n"
     ]
    }
   ],
   "source": [
    "# Filtrado de las parejas de Slorth con un origen experimental (BioGRID).\n",
    "SLslorthExp = SLslorth[SLslorth[5] == 'BioGRID']\n",
    "print(SLslorthExp)\n",
    "\n",
    "# Filtrado de las parejas de Slorth con un origen computacional.\n",
    "SLslorthComp = SLslorth[SLslorth[5] == 'Slorth']\n",
    "print(SLslorthComp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "      n1.name n2.name  score\n",
      "0       CHEK1    MTOR    3.0\n",
      "1         ADA   CHEK1    2.0\n",
      "2       BRCA1   PARP1    2.0\n",
      "3       BRCA2   PARP1    2.0\n",
      "4    C10orf76   PTAR1    2.0\n",
      "..        ...     ...    ...\n",
      "928    TUBA1A   YWHAZ    1.0\n",
      "929     VEGFA     VHL    1.0\n",
      "930      WEE1    WEE1    1.0\n",
      "931      WEE1     WRN    1.0\n",
      "932      WEE1   XRCC3    1.0\n",
      "\n",
      "[933 rows x 3 columns]\n",
      "       n1.name n2.name     score\n",
      "933      CDC27   MYLIP  0.735260\n",
      "934      CDC27  NCAPD2  0.853135\n",
      "935      CDC27  POLR3B  0.861576\n",
      "936      CDC27    CDH1  0.783209\n",
      "937      CDC27   VAMP3  0.706373\n",
      "...        ...     ...       ...\n",
      "518631     TPR   U2AF1  0.822218\n",
      "518632     TPR   VTI1A  0.814406\n",
      "518633    TSC2   U2AF1  0.816285\n",
      "518634    TSC2   VTI1A  0.855900\n",
      "518635    TSC2  ZBTB16  0.743188\n",
      "\n",
      "[517703 rows x 3 columns]\n"
     ]
    }
   ],
   "source": [
    "# Función que ajusta el formato de las parejas para la función SLcsv().\n",
    "def SLcolumns(SLdf):\n",
    "    SLdf = SLdf.iloc[:,[0,1,7]]\n",
    "    SLdf.columns = ['n1.name','n2.name','score']\n",
    "    return SLdf\n",
    "\n",
    "SLslorthExp = SLcolumns(SLslorthExp)\n",
    "print(SLslorthExp)\n",
    "\n",
    "SLslorthComp = SLcolumns(SLslorthComp)\n",
    "print(SLslorthComp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       n1.name n2.name     score\n",
      "934      CDC27  NCAPD2  0.853135\n",
      "935      CDC27  POLR3B  0.861576\n",
      "936      CDC27    CDH1  0.783209\n",
      "938      CDC27  ARID1B  0.757375\n",
      "939      CDC27  ANAPC4  0.898875\n",
      "...        ...     ...       ...\n",
      "518630     TPR    TSC2  0.823523\n",
      "518631     TPR   U2AF1  0.822218\n",
      "518632     TPR   VTI1A  0.814406\n",
      "518633    TSC2   U2AF1  0.816285\n",
      "518634    TSC2   VTI1A  0.855900\n",
      "\n",
      "[364720 rows x 3 columns]\n"
     ]
    }
   ],
   "source": [
    "# Filtrado de las fuentes computacionales de Slorth en base a su score de calidad\n",
    "# (>= 0.75 --- High Quality Score)\n",
    "SLslorthComp = SLslorthComp[SLslorthComp['score'] >= 0.75]\n",
    "print(SLslorthComp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------SL pairs for Essential PatuT Genes in SlorthExp:\n",
      "\n",
      "SL pairs in SlorthExp for PRMT3: No SL pairs Found\n",
      "\n",
      "SL pairs in SlorthExp for ASXL1: No SL pairs Found\n",
      "\n",
      "SL pairs in SlorthExp for HELLS: No SL pairs Found\n",
      "\n",
      "SL pairs in SlorthExp for SMARCE1: No SL pairs Found\n",
      "\n",
      "SL pairs in SlorthExp for KDM4B: No SL pairs Found\n",
      "\n",
      "SL pairs in SlorthExp for EPC2: No SL pairs Found\n",
      "\n",
      "-------SL pairs for Suppressor PatuT Genes in SlorthExp:\n",
      "\n",
      "SL pairs in SlorthExp for WHSC1: No SL pairs Found\n",
      "\n",
      "SL pairs in SlorthExp for SUV39H2: ['KRAS']\n",
      "\n",
      "-------SL pairs for Essential PatuS Genes in SlorthExp:\n",
      "\n",
      "SL pairs in SlorthExp for CHD7: No SL pairs Found\n",
      "\n",
      "SL pairs in SlorthExp for BRD4: ['CDC7', 'CDK9', 'CHEK1', 'CHEK2', 'DHFR', 'FGFR3', 'KDM5C', 'MAP2K1', 'PTAR1', 'PTEN', 'RB1', 'WEE1', 'BRCA1']\n",
      "\n",
      "SL pairs in SlorthExp for ING5: ['MAP2K1', 'TUBA1A', 'CDC7', 'CDK9']\n",
      "\n",
      "-------SL pairs for Suppressor PatuS Genes in SlorthExp:\n",
      "\n",
      "SL pairs in SlorthExp for PHF23: No SL pairs Found\n",
      "\n",
      "SL pairs in SlorthExp for UBE2A: No SL pairs Found\n",
      "\n"
     ]
    }
   ],
   "source": [
    "SLcsv(SLslorthExp,'SlorthExp')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------SL pairs for Essential PatuT Genes in SlorthComp:\n",
      "\n",
      "SL pairs in SlorthComp for PRMT3: ['DCTN1', 'PDLIM7']\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SL pairs in SlorthComp for ASXL1: ['CDH1', 'RHOA', 'POLA1', 'SMAD4', 'PHB', 'CANX', 'SIRT1', 'PDCD6IP', 'ACTB', 'UBE2K', 'CHD4', 'POLD1', 'TXN', 'HSPA5', 'ASF1B', 'HIPK2', 'MAPK14', 'RAC1', 'SIN3B', 'PRKCB', 'HSPB1', 'MTA2', 'KDM1A', 'NFE2L2', 'HNRNPA2B1', 'TSG101', 'BMI1', 'KEAP1', 'JUND', 'EZR', 'RBL2', 'JUNB', 'SKI', 'E2F1', 'POU2F1', 'SFPQ']\n",
      "\n",
      "SL pairs in SlorthComp for HELLS: ['PSEN1', 'MAP3K1', 'ACTB', 'HSPA8', 'PRKACA', 'UBE2K', 'APEX1', 'HSPA5', 'RPS6KA1', 'CSNK1A1', 'IKBKB', 'HIPK2', 'SUPT16H', 'AURKA', 'HSPB1', 'SMARCE1', 'KDM1A', 'NFE2L2', 'TNK2', 'TSG101', 'AHR', 'SFPQ', 'TNFAIP3']\n",
      "\n",
      "SL pairs in SlorthComp for SMARCE1: ['PTPN23', 'THOC1', 'PSEN1', 'XPO1', 'HUWE1', 'SNAP23', 'MAP3K1', 'JAK2', 'SMARCB1', 'GRAP2', 'SNW1', 'PSMB5', 'HCK', 'NOP56', 'POLA1', 'NUP93', 'RNF40', 'SNRNP70', 'FBL', 'CDK6', 'MET', 'UBE2R2', 'EIF3A', 'UBTF', 'RPS6KB1', 'UBE2D3', 'FBXW7', 'ARPC3', 'CDKN1B', 'RRP9', 'NCL', 'SPTBN1', 'SOS1', 'MSH6', 'RPF1', 'ARID1A', 'RPA2', 'TJP2', 'COPS5', 'TUBA1B', 'CNOT1', 'SNRPB2', 'UBA2', 'PIN1', 'GNL3L', 'CDC16', 'UBE2M', 'UBE4B', 'UBA1', 'DCTN4', 'KRAS', 'VHL', 'CDC73', 'CDCA8', 'APC', 'WDR36', 'CDK4', 'EPRS', 'CDK9', 'NR5A1', 'RAB1A', 'ITGAV', 'USP8', 'COPS4', 'NDUFA9', 'SNRPF', 'BRCA2', 'IGF1R', 'NDUFB10', 'NCOR1', 'COPS3', 'EIF4A3', 'SMAD4', 'BRD4', 'SAE1', 'CARM1', 'PSMB6', 'PSMA5', 'PARP1', 'HSPD1', 'FANCD2', 'CCNA2', 'RBMX', 'NOTCH1', 'ATM', 'CHEK1', 'ANAPC1', 'EEF1A1', 'BUB1B', 'KIT', 'BRAF', 'PINK1', 'RUNX1', 'NXF1', 'JAK1', 'RCHY1', 'PRKCD', 'PBRM1', 'COG5', 'DCAF13', 'TSC1', 'COPS2', 'CCT2', 'WEE1', 'PLK1', 'GATAD2A', 'COPS6', 'RFWD3', 'COG7', 'MAP2K1', 'BCL2', 'EXOSC10', 'RRM2', 'PTEN', 'QARS', 'TOMM20', 'RAB1B', 'TYMS', 'ERN1', 'EXOSC4', 'GAK', 'CALR', 'NOP10', 'IST1', 'CHEK2', 'EIF3C', 'EPOR', 'FANCA', 'IARS', 'PTPN1', 'HDAC2', 'AP2A1', 'PSMD12', 'COPS8', 'MTOR', 'TOP1', 'SNX2', 'PPP2R2A', 'PRKDC', 'COG8', 'UBA3', 'UBQLN4', 'WDHD1', 'UBE2L3', 'RPS24', 'CDC34', 'STX7', 'ARHGDIA', 'IPO9', 'PHB', 'MAP3K3', 'CANX', 'YARS', 'NUP98', 'DNAJA1', 'SIRT1', 'MCM4', 'NAPA', 'XPA', 'PDCD6IP', 'ACTB', 'XPC', 'TAF5L', 'HELZ2', 'HELLS', 'LARS', 'MCM3AP', 'AARS', 'RAB11A', 'DDX24', 'NEK2', 'TTK', 'DDX19A', 'SUGT1', 'CUL4A', 'PHF12', 'UBE2S', 'MAPK10', 'TTC1', 'PRKAR1A', 'ATP6V1B2', 'PSMB7', 'PSMA3', 'CDC7', 'PSMB4', 'FEN1', 'SGK1', 'RAN', 'PSMC3', 'NLK', 'RBBP5', 'MAP2K2', 'PSMD8', 'RASA1', 'SMARCC2', 'WWP2', 'PSMA6', 'MKNK1', 'USP46', 'PRDX1', 'FOXO1', 'TUBA3E', 'VDAC1', 'GGA2', 'TRMT1', 'BTRC', 'SRSF6', 'MLST8', 'BARD1', 'MAP3K4', 'SRSF4', 'RAD17', 'PSMC2', 'UBQLN1', 'TKT', 'UBR5', 'KAT5', 'PRMT1', 'RPS29', 'SSRP1', 'TAF9', 'H2AFX', 'RPS6KA3', 'ARHGEF2', 'CTBP2', 'PSMA8', 'HSPA8', 'EXOSC2', 'VRK1', 'NOLC1', 'CHMP5', 'PRKACB', 'CFL1', 'TIAL1', 'TUBB2A', 'ACTG1', 'HDAC5', 'RPL13A', 'MCM5', 'NCAPH', 'SNRNP200', 'RBM22', 'HCFC1', 'RAB11B', 'ARF6', 'PSMC1', 'CHD3', 'CHMP1A', 'RAD23B', 'COPB1', 'RPS7', 'DNM1', 'EFTUD2', 'VAV2', 'RPL36', 'RAC3', 'CUL5', 'PRKCI', 'ACTL6B', 'MAP2K6', 'TOP2A', 'TAF1', 'PFDN5', 'UBE2K', 'MYB', 'H3F3A', 'USP22', 'SENP2', 'EP400', 'EDF1', 'PPP2R5D', 'DDX56', 'CHD4', 'SRSF5', 'POLE3', 'CYCS', 'STRN4', 'UBE2N', 'AIMP1', 'RPS14', 'POLR3A', 'TERT', 'COPA', 'TXN', 'APEX1', 'CTBP1', 'UBE2V1', 'BECN1', 'RNF20', 'ESPL1', 'CALM1', 'EIF4E', 'ING1', 'HSPA2', 'HSP90AB1', 'CKS2', 'NDC80', 'CSTF2', 'CDC25B', 'PSMD7', 'ASF1B', 'HGS', 'EZH2', 'SEPT2', 'CDC25C', 'EXO1', 'PSMD9', 'LEO1', 'RPA1', 'MYBL2', 'STRIP1', 'CUL2', 'RAD23A', 'YWHAH', 'RPS6KA1', 'HSPA6', 'ACTA1', 'CSNK1D', 'BUB3', 'CKAP5', 'CCNB1', 'CLINT1', 'CDC23', 'HIST2H2AC', 'PDPK1', 'KAT2B', 'WWP1', 'USP34', 'RCC1', 'UBXN2B', 'TBP', 'DAD1', 'DNA2', 'EIF2B2', 'EIF3B', 'HAT1', 'MSH2', 'ASCC3', 'CSNK1A1', 'TGFBRAP1', 'EIF2S1', 'SETX', 'CDK7', 'DMAP1', 'ERCC4', 'SF1', 'MAD2L1', 'GGA1', 'SYVN1', 'SMC4', 'PPIF', 'PRKCA', 'VPS25', 'CDC45', 'GNAQ', 'ATR', 'YWHAQ', 'POLR2C', 'SSU72', 'ENO1', 'HERC2', 'GNA13', 'CDC6', 'STX4', 'KPNB1', 'MCM7', 'BLM', 'AMFR', 'PPIA', 'TUBA4A', 'UBE2B', 'CHUK', 'BUB1', 'COPB2', 'PSMC5', 'EIF4G1', 'PSMD3', 'AP2A2', 'PRPF6', 'UBE3A', 'HMGB1', 'EEF1G', 'RFC4', 'SSB', 'SEPT11', 'AP1B1', 'ASNA1', 'POLR1E', 'ING5', 'RANGAP1', 'NSFL1C', 'CEBPZ', 'UBE2V2', 'KHDRBS1', 'ING4', 'PAK1', 'RRP1B', 'PRKAR2A', 'NAP1L1', 'SLX1A', 'DCTN1', 'RAB7A', 'MAP3K2', 'STIP1', 'ORC1', 'PSMC6', 'NCAPG', 'H2AFV', 'SRCAP', 'SRSF9', 'SPATA5', 'ERCC2', 'RANBP2', 'VCP', 'DDX1', 'PSMD11', 'INPP5D', 'EIF4G2', 'PPID', 'EHMT2', 'CDC25A', 'NUP107', 'UBE2E3', 'PSMB10', 'SRSF1', 'RBMS2', 'FAF2', 'CHTF18', 'FZR1', 'GNL3', 'TCP1', 'AP2M1', 'UBL5', 'RAB8A', 'SHMT2', 'CNOT8', 'CSTF3', 'GSK3A', 'MAP3K5', 'CYC1', 'PRKCE', 'TTC37', 'NAPB', 'TXNL1', 'PPM1B', 'WWOX', 'MAPK14', 'ACTR6', 'H2AFY', 'TOMM70', 'DYNLL1', 'WRNIP1', 'SMG1', 'BUD31', 'NEDD8', 'IGBP1', 'TOP2B', 'SCFD1', 'PRDX2', 'PSMA7', 'LRRK2', 'NCBP2', 'ITCH', 'RUVBL1', 'PSMD6', 'SENP3', 'GINS1', 'TCEA1', 'MORF4L1', 'SUPT16H', 'NCOR2', 'CDK5', 'NFYB', 'PPIH', 'LSM7', 'STRN3', 'RAB5A', 'SMC3', 'SF3A1', 'RPN1', 'SEC13', 'MLH1', 'SEPT7', 'ATP5O', 'POLR2E', 'ZMYND8', 'PAK2', 'RFC3', 'NAP1L4', 'POLI', 'REV1', 'YEATS4', 'CCNB2', 'TUBB6', 'UBE2E1', 'UBE2E2', 'MAPK3', 'ISG15', 'EBNA1BP2', 'SOD1', 'TBL1X', 'EPS15', 'KPNA1', 'RPL37', 'HRAS', 'RPS4Y1', 'USP14', 'PRPF19', 'TRIM23', 'VAMP8', 'USP7', 'POLH', 'HIST3H3', 'ACTL6A', 'VPS26A', 'SMARCC1', 'CDC20', 'UNC45A', 'PPP2CA', 'PIAS4', 'PSMD14', 'YWHAE', 'EEF2', 'PSMD4', 'MAP1LC3A', 'RRAGA', 'SMC6', 'TCEB1', 'MAP2K7', 'CSTF2T', 'FAF1', 'PEX5', 'CKS1B', 'CUL4B', 'USP3', 'CHAF1B', 'RPS21', 'STX5', 'HNRNPM', 'TRAP1', 'VAMP2', 'PEX14', 'PPP3CA', 'HIST1H3J', 'FOXO3', 'TBK1', 'TOM1L2', 'RICTOR', 'CCNE1', 'PLAA', 'MCM3', 'SIN3B', 'BCR', 'PFDN2', 'PRKAA1', 'BYSL', 'TPT1', 'RPSA', 'YWHAG', 'ASF1A', 'VPS4B', 'STAM2', 'AURKB', 'PSMB2', 'YKT6', 'EIF3F', 'RAE1', 'SFN', 'UBQLN2', 'SMARCA5', 'PKMYT1', 'USP9X', 'CSNK1E', 'PSMA2', 'ATXN2', 'TUBA1C', 'TUBA1A', 'STRN', 'USP10', 'HSP90B1', 'SGTA', 'UBE2A', 'PIAS2', 'CDK8', 'PRKCB', 'H2AFZ', 'PPP4C', 'PPIB', 'UBE2G2', 'FOXO4', 'AURKA', 'SUV39H1', 'PRKAB2', 'DDX3X', 'SMURF1', 'PPP2CB', 'DNM2', 'SMARCA1', 'WDR48', 'SUMO3', 'DDX39A', 'PHB2', 'TBL1XR1', 'RPS10', 'WRN', 'MORF4L2', 'SENP1', 'DDOST', 'ACTN2', 'MTA1', 'INSR', 'RBPJ', 'OGT', 'PSEN2', 'VPS36', 'HSPB1', 'PDS5A', 'MTA2', 'DVL3', 'GTF3C5', 'ERBB4', 'ATP5B', 'SMAD1', 'SMAD5', 'ITPR1', 'SNF8', 'CBLC', 'SHOC2', 'MED23', 'ACTN4', 'ARPC4', 'RARS', 'SMAD2', 'EXOSC9', 'PDS5B', 'SMAD7', 'GPN1', 'CCT5', 'CHMP4B', 'PARD6G', 'PARD6A', 'SMAD6', 'CBLB', 'FGFR1', 'ATP5F1', 'SF3B1', 'RPL3L', 'PPP2R2C', 'GARS', 'MED12', 'ATF6', 'SUPT5H', 'ETS2', 'AXIN1', 'EIF3M', 'RNF2', 'NFKB2', 'ETS1', 'CTNND1', 'ETF1', 'IRS1', 'PRMT5', 'NFKBIA', 'PSME3', 'CDT1', 'G3BP1', 'PSME2', 'NFE2L2', 'CDC37', 'SPTAN1', 'TRAF4', 'PIK3C2B', 'RPTOR', 'DTL', 'ARRB1', 'SET', 'LATS1', 'ZAP70', 'MED13', 'SMC5', 'IK', 'ALYREF', 'SIAH1', 'HNRNPA2B1', 'DIS3', 'EIF4EBP1', 'STK4', 'RBL1', 'RUVBL2', 'STK38', 'DHX15', 'MARK3', 'SOCS3', 'DYNC1H1', 'RIN1', 'SND1', 'SRSF3', 'THRB', 'PTK2', 'BIRC5', 'PAPOLA', 'LMNA', 'YY1', 'ANAPC2', 'MAPKAP1', 'RAD21', 'CBX1', 'MAP3K7', 'DMD', 'ANAPC11', 'RPS19', 'TCEB2', 'ZC3H14', 'RARA', 'UTRN', 'CCND1', 'USP32', 'MAPT', 'ARNTL', 'HNRNPA0', 'SNAI1', 'KDR', 'SKIL', 'PARK2', 'FOS', 'PLCG2', 'MSN', 'SMARCA2', 'ARRB2', 'CCNG1', 'MEN1', 'CSE1L', 'RYBP', 'E2F4', 'CBFA2T3', 'TGFB1', 'MAP3K11', 'HDLBP', 'ITGB1', 'CTTN', 'LYN', 'EGLN3', 'SYK', 'KMT5A', 'PDGFRB', 'XRCC6', 'DOK1', 'SAP18', 'TRIM28', 'SLC9A3R1', 'FBXO11', 'STAG2', 'MAP1B', 'LATS2', 'SH3KBP1', 'TGFBR1', 'HNRNPH3', 'CBX3', 'CHAF1A', 'ATP5A1', 'RBX1', 'DARS', 'NOTCH3', 'HTRA2', 'ATG3', 'KLF5', 'STAU1', 'PSMA1', 'DYNLT3', 'SMARCD2', 'ADRB2', 'XIAP', 'CLTC', 'STAT5A', 'HNRNPAB', 'BIRC2', 'TP53BP2', 'PPARGC1A', 'CASP7', 'YAP1', 'ACIN1', 'RAP1A', 'ANAPC5', 'CLSPN', 'SERBP1', 'SMU1', 'CSK', 'LMNB1', 'DHX9', 'DCTN2', 'PXN', 'CCT7', 'PPP1CA', 'SMARCD3', 'TSG101', 'CLN3', 'FLNC', 'SP3', 'CLNS1A', 'BIRC6', 'TCF4', 'AHR', 'ACACB', 'NEDD9', 'HUS1', 'RELB', 'RUNX2', 'BMI1', 'PSME1', 'SUZ12', 'CD2AP', 'KEAP1', 'JUND', 'SNU13', 'TIMM50', 'EZR', 'MYOD1', 'HNRNPH1', 'PDLIM7', 'CASP9', 'ATP6V1C1', 'LCK', 'MYH9', 'NFKB1', 'GAB1', 'CCND3', 'PARK7', 'RBL2', 'WDR74', 'YES1', 'RUNX1T1', 'STAG1', 'MYH10', 'SCAF4', 'NF2', 'VDR', 'TYK2', 'POLDIP3', 'SH3GL2', 'CCND2', 'EIF4A1', 'TRIM24', 'JUNB', 'DLG1', 'ID2', 'HNRNPD', 'PIK3R2', 'HNRNPA1', 'HNRNPF', 'SKI', 'RBBP7', 'DDB1', 'STAT5B', 'TOPORS', 'SREBF2', 'MAPK8IP1', 'INO80', 'TADA3', 'CASP3', 'HNRNPA3', 'TRIM55', 'HNRNPDL', 'ERCC3', 'JAK3', 'POLR1B', 'E2F1', 'FOXG1', 'DERL1', 'POU2F1', 'WAC', 'ORC5', 'RET', 'CBX5', 'SNRPA1', 'FAM96A', 'TFDP1', 'SKP2', 'ASH2L', 'MYD88', 'ETV6', 'SF3A3', 'SNRPD3', 'PYM1', 'AGO1', 'STAT6', 'APP', 'AEBP2', 'DEK', 'SIAH2', 'TAF4', 'NBN', 'AGO2', 'SOCS1', 'BAP1', 'BCL6', 'CDKN1A', 'CEBPA', 'CNOT3', 'DDB2', 'DDIT3', 'DNMT3A', 'EIF3E', 'ERCC5', 'FANCC', 'FANCG', 'FIP1L1', 'FUBP1', 'FUS', 'HMGA1', 'IKZF1', 'KLF4', 'MALT1', 'MDM4', 'MLLT1', 'MLLT3', 'NCOA1', 'NCOA2', 'NONO', 'NPM1', 'NTRK1', 'PCBP1', 'PML', 'POLE', 'PPARG', 'PPP2R1A', 'SFPQ', 'TAL1', 'TGFBR2', 'TNFAIP3', 'TSC2', 'U2AF1', 'VTI1A', 'ZBTB16', 'CDC27', 'BRCA1', 'POLR3B', 'CDH1', 'CASP8', 'KARS', 'RHOA', 'FGFR3', 'UBE2D1', 'USP13', 'THOC3', 'MEF2A', 'CUL1', 'PSMC4', 'PRKACA', 'CDC42', 'POLD1', 'HSPA5', 'CUL3', 'RPS6KA2', 'UBE3C', 'POLR1A', 'RIOK2', 'PABPC1', 'FBXW11', 'PSMB1', 'RFC1', 'NOP58', 'HIPK2', 'MAP2K4', 'NEDD4', 'MCM2', 'DCUN1D1', 'NEDD4L', 'RAD18', 'CREBBP', 'RECQL', 'HDAC7', 'RALBP1', 'ELAVL1', 'UPF1', 'PSMA4', 'CFTR', 'HDAC4', 'UTP18', 'ACTN1', 'POLR2B', 'CTNNA1', 'KDM1A', 'PTBP1', 'ZRANB1', 'FYN', 'TNK2', 'UFD1L', 'MARK2', 'SMARCD1', 'BIRC3', 'SLC25A5', 'KMT2C', 'YBX1', 'U2AF2', 'SREBF1', 'BCL3', 'THRAP3', 'TP63', 'TPR']\n",
      "\n",
      "SL pairs in SlorthComp for KDM4B: ['KRAS', 'CASP7', 'SIAH2', 'IDE', 'BCLAF1']\n",
      "\n",
      "SL pairs in SlorthComp for EPC2: ['SRCAP', 'TAF1L', 'EPC1']\n",
      "\n",
      "-------SL pairs for Suppressor PatuT Genes in SlorthComp:\n",
      "\n",
      "SL pairs in SlorthComp for WHSC1: ['FGFR2', 'PRKACA', 'HDAC9', 'TCF3', 'KDM1A', 'PRDM1']\n",
      "\n",
      "SL pairs in SlorthComp for SUV39H2: No SL pairs Found\n",
      "\n",
      "-------SL pairs for Essential PatuS Genes in SlorthComp:\n",
      "\n",
      "SL pairs in SlorthComp for CHD7: No SL pairs Found\n",
      "\n",
      "SL pairs in SlorthComp for BRD4: ['FANCD2', 'FOXO1', 'PRKCI', 'SF1', 'AURKB', 'PRKCB', 'OGT', 'BMI1', 'SREBF2', 'MYD88', 'DDIT3', 'CDC27', 'CDH1', 'CTDP1', 'CASP8', 'FGFR2', 'UBE2D1', 'GSK3B', 'XPO1', 'JAK2', 'ABL1', 'SMARCB1', 'MAPK1', 'POLA1', 'UBE2I', 'SNRNP70', 'UBE2D3', 'FBXW7', 'CDKN1B', 'NCL', 'HDAC1', 'RPA2', 'COPS5', 'ZC3H13', 'CDK2', 'PIN1', 'UBA1', 'RAF1', 'VHL', 'APC', 'CDK4', 'MYC', 'NCOR1', 'EIF4A3', 'SMAD4', 'ERBB2', 'CDC34', 'SIRT1', 'MCM4', 'CUL4A', 'CDC7', 'RAN', 'PCNA', 'BARD1', 'CSNK2A1', 'UBQLN1', 'PRMT1', 'HSPA8', 'CUL1', 'UBE2D2', 'RAD23B', 'EFTUD2', 'PRKACA', 'TOP2A', 'UBE2K', 'POLD1', 'APEX1', 'HSP90AB1', 'SUMO1', 'ASF1B', 'EZH2', 'RPA1', 'CUL2', 'RPS6KA1', 'CCNB1', 'KAT2B', 'WWP1', 'TBP', 'MSH2', 'PABPC1', 'CDC45', 'CDC6', 'KPNB1', 'UBE2B', 'EIF4G1', 'PRPF6', 'UBE3A', 'KHDRBS1', 'IKBKB', 'ORC1', 'HIPK2', 'ERCC2', 'SRSF1', 'NEDD4', 'MAPK14', 'DYNLL1', 'MCM2', 'NEDD8', 'NCBP1', 'TOP2B', 'NEDD4L', 'NCBP2', 'ITCH', 'SUPT16H', 'RAD18', 'POLR2E', 'CREBBP', 'RECQL', 'MAPK3', 'PRPF19', 'ACTL6A', 'CDC20', 'PPP2CA', 'PIAS4', 'RAC1', 'FOXO3', 'CCNE1', 'SIN3B', 'UPF1', 'USP10', 'PIAS2', 'FKBP1A', 'AURKA', 'SUV39H1', 'CFTR', 'MORF4L2', 'CBL', 'HSPB1', 'SMAD7', 'CBLB', 'POLR2B', 'SMARCE1', 'TCF3', 'AXIN1', 'RNF2', 'ETS1', 'KDM1A', 'NFKBIA', 'PSME3', 'NFE2L2', 'PTBP1', 'ARRB1', 'FYN', 'YY1', 'MAP3K7', 'RARA', 'CCND1', 'KDR', 'SMARCA2', 'ARRB2', 'TNK2', 'PDGFRB', 'TRIM28', 'SLC9A3R1', 'RBX1', 'HIF1A', 'XIAP', 'STAT5A', 'BIRC2', 'YAP1', 'DHX9', 'PXN', 'AHR', 'RUNX2', 'PLCG1', 'KEAP1', 'BIRC3', 'SKP1', 'YBX1', 'MYOD1', 'NFKB1', 'PARK7', 'RBL2', 'VDR', 'HNRNPD', 'HNRNPA1', 'RBBP7', 'U2AF2', 'E2F1', 'CBX5', 'CREB1', 'SREBF1', 'DDX17', 'AGO2', 'BCL3', 'BCL6', 'CDKN1A', 'ESR1', 'MLLT1', 'PML', 'PPARG', 'SFPQ', 'TNFAIP3', 'TPR', 'TSC2']\n",
      "\n",
      "SL pairs in SlorthComp for ING5: ['MAP2K1', 'CALR', 'WDHD1', 'CTBP2', 'CHD3', 'MAP3K2', 'POLH', 'CKS1B', 'AURKB', 'CTNND1', 'TCF4', 'PDLIM7', 'NF2', 'JUNB', 'TFDP1', 'MYD88', 'CEBPA', 'DDIT3', 'POLE', 'PSEN1', 'XPO1', 'JAK2', 'SMARCB1', 'POLA1', 'UBE2D3', 'CDKN1B', 'RPA2', 'CDK2', 'UBA2', 'PIN1', 'VHL', 'APC', 'CDK4', 'EIF4A3', 'SMAD4', 'ERBB2', 'AKT1', 'PARP1', 'FANCD2', 'NOTCH1', 'ATM', 'CHEK1', 'BRAF', 'NXF1', 'RCHY1', 'PRKCD', 'PLK1', 'MAPK8', 'PHB', 'SIRT1', 'ACTB', 'SUGT1', 'SGK1', 'RAN', 'RBBP5', 'MAPK9', 'FOXO1', 'BARD1', 'RAD17', 'PRMT1', 'SSRP1', 'HSPA8', 'KAT7', 'PRKCI', 'MAP2K6', 'TOP2A', 'TAF1', 'UBE2K', 'H3F3A', 'POLD1', 'TERT', 'TXN', 'APEX1', 'HSPA5', 'CTBP1', 'CUL3', 'ING1', 'ASF1B', 'EZH2', 'TAF5', 'RPS6KA1', 'CSNK1D', 'CCNB1', 'KAT2B', 'KAT6A', 'DNA2', 'HAT1', 'MSH2', 'POLR1A', 'EIF2S1', 'RAD51', 'PIAS1', 'SF1', 'PRKCA', 'CDC6', 'KPNB1', 'TP53I3', 'KHDRBS1', 'ING4', 'HIPK2', 'ERCC2', 'SRSF1', 'TAF6', 'NEDD4', 'MAPK14', 'DYNLL1', 'SMG1', 'NFYA', 'ITCH', 'SUPT16H', 'CDK5', 'NFYB', 'SMC3', 'RAD18', 'MAPK3', 'HIST3H3', 'STK11', 'ACTL6A', 'TCEB1', 'MAP2K7', 'STX5', 'CCNE1', 'SIN3B', 'SMARCA5', 'UPF1', 'USP10', 'PIAS2', 'PRKCB', 'AURKA', 'EIF2AK2', 'WDR48', 'WRN', 'HDAC4', 'MORF4L2', 'OGT', 'HSPB1', 'ATP5B', 'CTNNB1', 'ACTN4', 'SMAD7', 'FGFR1', 'ATP5F1', 'SMARCE1', 'RNF2', 'ETS1', 'KDM1A', 'CDT1', 'NFE2L2', 'RPTOR', 'DTL', 'ARRB1', 'STK4', 'MAP3K7', 'TCEB2', 'RARA', 'UTRN', 'CCND1', 'EPC1', 'KDR', 'SMARCA2', 'ARRB2', 'TNK2', 'TGFB1', 'PDGFRB', 'SAP18', 'TRIM28', 'SLC9A3R1', 'HIF1A', 'XIAP', 'UFD1L', 'TP53BP2', 'YAP1', 'DHX9', 'PXN', 'AHR', 'BMI1', 'KEAP1', 'JUND', 'YBX1', 'MYOD1', 'EPAS1', 'CASP9', 'RBL2', 'VDR', 'TRIM24', 'RBBP7', 'U2AF2', 'INO80', 'CASP3', 'ERCC3', 'E2F1', 'POU2F1', 'CBX5', 'CREB1', 'DDX17', 'BCL6', 'NCOA1', 'PML', 'PPARG', 'TAL1', 'TNFAIP3', 'TPR']\n",
      "\n",
      "-------SL pairs for Suppressor PatuS Genes in SlorthComp:\n",
      "\n",
      "SL pairs in SlorthComp for PHF23: No SL pairs Found\n",
      "\n",
      "SL pairs in SlorthComp for UBE2A: ['PSEN1', 'GSK3B', 'XPO1', 'CYLD', 'HUWE1', 'SNAP23', 'MAP3K1', 'JAK2', 'ABL1', 'SMARCB1', 'MAPK1', 'GRAP2', 'PSMB5', 'HCK', 'POLA1', 'MID1', 'NAA10', 'NUP93', 'UBE2I', 'GSPT1', 'RNF40', 'SNRNP70', 'CDK6', 'MET', 'UBE2R2', 'EIF3A', 'UBTF', 'RPS6KB1', 'GOSR1', 'UBE2D3', 'FBXW7', 'ARPC3', 'CDKN1B', 'HSPA9', 'RRP9', 'NCL', 'SPTBN1', 'SOS1', 'MSH6', 'RPF1', 'RPA2', 'TJP2', 'COPS5', 'CDK2', 'TUBA1B', 'CNOT1', 'SNRPB2', 'UBA2', 'PIN1', 'GNL3L', 'CDC16', 'UBE2M', 'UBE4B', 'UBA1', 'RAF1', 'DCTN4', 'KRAS', 'VHL', 'CDC73', 'CDCA8', 'APC', 'WDR36', 'CDK4', 'EPRS', 'CDK9', 'MYC', 'RAB1A', 'ACTR2', 'ITGAV', 'USP8', 'COPS4', 'NDUFA9', 'SNRPF', 'BRCA2', 'IGF1R', 'NDUFB10', 'NCOR1', 'COPS3', 'EIF4A3', 'SMAD4', 'ERBB2', 'SAE1', 'CARM1', 'PSMB6', 'PSMA5', 'PARP1', 'HSPD1', 'FANCD2', 'CCNA2', 'MARCH6', 'RBMX', 'NOTCH1', 'ATM', 'CHEK1', 'UBE3B', 'ANAPC1', 'EEF1A1', 'BUB1B', 'KIT', 'BRAF', 'PINK1', 'RUNX1', 'SHC1', 'NXF1', 'JAK1', 'ARPC2', 'RCHY1', 'PRKCD', 'PBRM1', 'DCAF13', 'TSC1', 'COPS2', 'CCT2', 'WEE1', 'PLK1', 'GATAD2A', 'COPS6', 'RFWD3', 'MAP2K1', 'BCL2', 'EXOSC10', 'PTEN', 'QARS', 'RAB1B', 'UBE2O', 'TYMS', 'ERN1', 'EXOSC4', 'GAK', 'CALR', 'SETD2', 'IST1', 'CHEK2', 'EIF3C', 'EPOR', 'FANCA', 'IARS', 'HDAC2', 'AP2A1', 'PSMD12', 'COPS8', 'MTOR', 'TOP1', 'SNX2', 'PPP2R2A', 'PRKDC', 'UBA3', 'UBQLN4', 'WDHD1', 'UBE2L3', 'CDC34', 'STX7', 'MAPK8', 'ARHGDIA', 'IPO9', 'UBE2G1', 'PHB', 'MAP3K3', 'CANX', 'YARS', 'NUP98', 'DNAJA1', 'SIRT1', 'MCM4', 'NAPA', 'XPA', 'PDCD6IP', 'XPC', 'TAF5L', 'HELZ2', 'LARS', 'DLAT', 'MCM3AP', 'AARS', 'RAB11A', 'ATP6V1B1', 'DDX24', 'NEK2', 'TTK', 'SUGT1', 'CUL4A', 'GTF2H2C', 'UBE2S', 'MAPK10', 'TTC1', 'PRKAR1A', 'ATP6V1B2', 'PSMB7', 'PSMA3', 'CDC7', 'PSMB4', 'FEN1', 'SGK1', 'RAN', 'RAB25', 'PSMC3', 'NLK', 'RBBP5', 'MAP2K2', 'PSMD8', 'SMARCC2', 'WWP2', 'PSMA6', 'MKNK1', 'USP46', 'PRDX1', 'FOXO1', 'TUBA3E', 'VDAC1', 'GGA2', 'TRMT1', 'BTRC', 'SRSF6', 'MLST8', 'BARD1', 'MAP3K4', 'SRSF4', 'RAD17', 'PSMC2', 'UBQLN1', 'UBR5', 'KAT5', 'PRMT1', 'RPS29', 'SSRP1', 'HDAC3', 'BRSK1', 'TAF9', 'H2AFX', 'RPS6KA3', 'FASN', 'ARHGEF2', 'CTBP2', 'PSMA8', 'HSPA8', 'EXOSC2', 'NOLC1', 'CHMP5', 'PRKACB', 'CFL1', 'TUBB2A', 'ACTG1', 'RPL13A', 'HSPA4', 'MCM5', 'NCAPH', 'SNRNP200', 'RBM22', 'HCFC1', 'CHMP1B', 'RAB11B', 'ARF6', 'UBE2D2', 'PSMC1', 'CHD3', 'CHMP1A', 'RAD23B', 'COPB1', 'DNM1', 'EFTUD2', 'VAV2', 'RPL36', 'RAC3', 'CUL5', 'PRKCI', 'MAP2K6', 'TOP2A', 'TAF1', 'PFDN5', 'UBE2K', 'MYB', 'YWHAZ', 'H3F3A', 'USP22', 'SENP2', 'EP400', 'PPP2R5D', 'DDX56', 'CHD4', 'MAP1LC3C', 'SRSF5', 'POLE3', 'CYCS', 'STRN4', 'UBE2N', 'AIMP1', 'RPS14', 'POLR3A', 'TERT', 'COPA', 'TXN', 'APEX1', 'PPP6R1', 'CTBP1', 'UBE2V1', 'BECN1', 'RNF20', 'ESPL1', 'CALM1', 'EIF4E', 'YWHAB', 'ING1', 'HSPA2', 'SUCLA2', 'HSP90AB1', 'PC', 'CKS2', 'NDC80', 'CSTF2', 'CDC25B', 'PSMD7', 'ASF1B', 'HGS', 'EZH2', 'PPP2R5C', 'SEPT2', 'CDC25C', 'TAF5', 'EXO1', 'PSMD9', 'LEO1', 'STRIP1', 'CUL2', 'RAD23A', 'YWHAH', 'CALM2', 'RPS6KA1', 'HSPA6', 'ACTA1', 'CSNK1D', 'BUB3', 'TUBG1', 'CCNB1', 'CLINT1', 'CDC23', 'HIST2H2AC', 'PDPK1', 'KAT2B', 'WWP1', 'USP34', 'RCC1', 'TRIB3', 'UBXN2B', 'TBP', 'DNA2', 'EIF2B2', 'EIF3B', 'HAT1', 'MSH2', 'ASCC3', 'CSNK1A1', 'RPL22L1', 'TGFBRAP1', 'PIM1', 'EIF2S1', 'ERCC4', 'SF1', 'MAD2L1', 'GGA1', 'SYVN1', 'SMC4', 'PPIF', 'PRKCA', 'VPS25', 'CDC45', 'GNAQ', 'ATR', 'YWHAQ', 'POLR2C', 'SEC23A', 'HERC2', 'CDC6', 'STX4', 'KPNB1', 'MCM7', 'BLM', 'AMFR', 'PTRH2', 'TUBA4A', 'AKT2', 'UBE2B', 'CHUK', 'BUB1', 'COPB2', 'RPS25', 'PSMC5', 'EIF4G1', 'PSMD3', 'AP2A2', 'PRPF6', 'UBE3A', 'GCN1', 'GRK2', 'ACTC1', 'EEF1G', 'RFC4', 'SSB', 'SEPT11', 'AP1B1', 'ASNA1', 'PPP2R5E', 'POLR1E', 'VAV1', 'RANGAP1', 'NSFL1C', 'CEBPZ', 'UBE2V2', 'KHDRBS1', 'IKBKB', 'RRP1B', 'PRKAR2A', 'NAP1L1', 'SLX1A', 'DCTN1', 'GABARAP', 'ARFGAP3', 'MAP3K2', 'STIP1', 'ORC1', 'PSMC6', 'NCAPG', 'H2AFV', 'SRCAP', 'SRSF9', 'SPATA5', 'ERCC2', 'RANBP2', 'VCP', 'PSMD11', 'TOM1L1', 'ARCN1', 'INPP5D', 'EIF4G2', 'PPID', 'EHMT2', 'CDC25A', 'NUP107', 'UBE2E3', 'PSMB10', 'SRSF1', 'FAF2', 'CHTF18', 'FZR1', 'TCP1', 'AP2M1', 'UBL5', 'RAB8A', 'SHMT2', 'CNOT8', 'AHSA1', 'CSTF3', 'LNPEP', 'TAF6', 'ARHGAP32', 'GSK3A', 'MAP3K5', 'PRKCE', 'TTC37', 'NAPB', 'ITSN1', 'TXNL1', 'PPM1B', 'WWOX', 'MAPK14', 'SIN3A', 'OLA1', 'DYNLL1', 'MEF2C', 'WRNIP1', 'SMG1', 'NEDD8', 'IGBP1', 'NCBP1', 'SEH1L', 'CAPZA1', 'SCFD1', 'MAPRE1', 'PRDX2', 'PSMA7', 'LRRK2', 'NCBP2', 'ITCH', 'RUVBL1', 'PSMD6', 'SENP3', 'GINS1', 'UBE2H', 'TCEA1', 'MORF4L1', 'SUPT16H', 'CDK5', 'NFYB', 'PPIH', 'LSM7', 'DDX58', 'CDK1', 'MOV10', 'MEF2D', 'RABGEF1', 'RAB5A', 'SMC3', 'SF3A1', 'RPN1', 'SEC13', 'SEPT7', 'POLR2E', 'UBR1', 'ABCD3', 'PAK2', 'RFC3', 'NAP1L4', 'REV1', 'CCNB2', 'TUBB6', 'UBE2E1', 'UBE2E2', 'MAPK3', 'ISG15', 'EBNA1BP2', 'SOD1', 'EPS15', 'KPNA1', 'HRAS', 'USP14', 'PRPF19', 'HIP1', 'VAMP8', 'USP7', 'POLH', 'HIST3H3', 'STK11', 'ACTL6A', 'VPS26A', 'SMARCC1', 'CDC20', 'UNC45A', 'HMGCS1', 'PPP2CA', 'PIAS4', 'PSMD14', 'YWHAE', 'EEF2', 'PSMD4', 'VAPB', 'MAP1LC3A', 'RRAGA', 'SMC6', 'TCEB1', 'CSTF2T', 'FAF1', 'PEX5', 'CKS1B', 'CUL4B', 'USP3', 'CHAF1B', 'RPS21', 'STX5', 'HNRNPM', 'GAPDH', 'RAC1', 'TLN1', 'VAMP2', 'PPP3CA', 'VPS4A', 'HIST1H3J', 'FOXO3', 'TBK1', 'TOM1L2', 'PRKAG2', 'RICTOR', 'CCNE1', 'PLAA', 'MCM3', 'SIN3B', 'BCR', 'ATP6V1A', 'PRKAA1', 'BYSL', 'RPSA', 'YWHAG', 'VPS4B', 'STAM2', 'AURKB', 'UBXN7', 'PSMB2', 'YKT6', 'EIF3F', 'RAE1', 'SFN', 'UBQLN2', 'NARS', 'SMARCA5', 'PKMYT1', 'USP9X', 'CSNK1E', 'ORC2', 'PSMA2', 'TUBA1A', 'USP10', 'HSP90B1', 'CHP2', 'ATP6V0A2', 'SGTA', 'PIAS2', 'CCNA1', 'CDK8', 'PRKCB', 'KIF21A', 'H2AFZ', 'FKBP1A', 'UBE2G2', 'FOXO4', 'AURKA', 'SUV39H1', 'PRKAB2', 'DDX3X', 'SMURF1', 'PPP2CB', 'DNM2', 'SMARCA1', 'WDR48', 'COPG1', 'SUMO3', 'TBL1XR1', 'RPS10', 'WRN', 'IPO7', 'XPOT', 'MORF4L2', 'SENP1', 'MTA1', 'INSR', 'CBL', 'CAV1', 'OGT', 'PSEN2', 'VPS36', 'HSPB1', 'PDS5A', 'MTA2', 'DVL3', 'ERBB4', 'RAD1', 'ATP5B', 'SMAD1', 'SMAD5', 'ITPR1', 'CBLC', 'MED23', 'ACTN4', 'ARPC4', 'RARS', 'SMAD2', 'PDS5B', 'SMAD7', 'GPN1', 'CCT5', 'CHMP4B', 'PARD6G', 'SMAD3', 'PARD6A', 'SMAD6', 'CBLB', 'FGFR1', 'ATP5F1', 'SF3B1', 'GARS', 'MED12', 'SAV1', 'PIK3R3', 'ACTR3', 'SUPT5H', 'AXIN1', 'EIF3M', 'RNF2', 'ETS1', 'CTNND1', 'IRS1', 'PRMT5', 'TGFB3', 'NFKBIA', 'PSME3', 'CDT1', 'NANOS1', 'G3BP1', 'PSME2', 'NFE2L2', 'CDC37', 'SPTAN1', 'PIK3CD', 'EGLN1', 'RPTOR', 'DTL', 'ARRB1', 'SET', 'LATS1', 'ZAP70', 'NUP85', 'TUBGCP3', 'PDE4D', 'IK', 'ALYREF', 'SIAH1', 'THRA', 'HNRNPA2B1', 'DIS3', 'EIF4EBP1', 'RLIM', 'RBL1', 'RUVBL2', 'DHX15', 'SOCS3', 'DYNC1H1', 'RIN1', 'SND1', 'SRSF3', 'IDE', 'THRB', 'PTK2', 'BIRC5', 'PAPOLA', 'LMNA', 'YY1', 'ANAPC2', 'TGFB1I1', 'MAPKAP1', 'RAD21', 'TAOK1', 'SPRY2', 'CBX1', 'UBA7', 'MAP3K7', 'DMD', 'ANAPC11', 'RPS19', 'SP1', 'TCEB2', 'ZC3H14', 'RARA', 'UTRN', 'CCND1', 'RAPGEF1', 'USP32', 'MAPT', 'ARNTL', 'HNRNPA0', 'SNAI1', 'KDR', 'SKIL', 'PARK2', 'UVRAG', 'FOS', 'GMPS', 'PLCG2', 'MSN', 'SMARCA2', 'ARRB2', 'MEN1', 'CSE1L', 'RYBP', 'E2F4', 'TGFB1', 'MAP3K11', 'HDLBP', 'ITGB1', 'CTTN', 'ABI2', 'LYN', 'EGLN3', 'SYK', 'PDGFRB', 'XRCC6', 'SAP18', 'TRIM28', 'SLC9A3R1', 'FBXO11', 'STAG2', 'MAP1B', 'LATS2', 'SH3KBP1', 'TGFBR1', 'RELA', 'PDLIM5', 'CBX3', 'CHAF1A', 'NPHS1', 'ATP5A1', 'RBX1', 'DARS', 'HTRA2', 'ATG3', 'KLF5', 'STAU1', 'HIF1A', 'PSMA1', 'DYNLT3', 'SMARCD2', 'ADRB2', 'XIAP', 'CLTC', 'STAT5A', 'HNRNPAB', 'BIRC2', 'TP53BP2', 'KLC4', 'PPARGC1A', 'CASP7', 'SEC61B', 'YAP1', 'ACIN1', 'RAP1A', 'ITK', 'ANAPC5', 'CLSPN', 'SERBP1', 'SMU1', 'CSK', 'LMNB1', 'MARK1', 'DHX9', 'DCTN2', 'UBR4', 'RBM27', 'PXN', 'CCT7', 'PPP1CA', 'CLN3', 'FLNC', 'SP3', 'TRAF6', 'G3BP2', 'BIRC6', 'TCF4', 'AHR', 'NEDD9', 'HUS1', 'RELB', 'ATAD5', 'RUNX2', 'BMI1', 'PSME1', 'ARAF', 'PLCG1', 'CD2AP', 'KEAP1', 'SKP1', 'JUND', 'SNU13', 'TIMM50', 'EZR', 'MYOD1', 'HNRNPH1', 'PDLIM7', 'EPAS1', 'CASP9', 'ATP6V1C1', 'LCK', 'MYH9', 'NFKB1', 'GAB1', 'CCND3', 'MFN2', 'PARK7', 'RBL2', 'WDR74', 'YES1', 'RUNX1T1', 'STAG1', 'MYH10', 'NF2', 'VDR', 'POLDIP3', 'SH3GL2', 'CCND2', 'EIF4A1', 'TRIM24', 'JUNB', 'ID2', 'HNRNPD', 'PIK3R2', 'HNRNPA1', 'HNRNPF', 'ECT2', 'SKI', 'RBBP7', 'DDB1', 'STAT5B', 'TOPORS', 'SREBF2', 'MAPK8IP1', 'INO80', 'TADA3', 'CASP3', 'TRIM55', 'HNRNPDL', 'ERCC3', 'MRPL4', 'JAK3', 'AGO3', 'POLR1B', 'E2F1', 'LDB3', 'GIPC1', 'SRRM1', 'DERL1', 'POU2F1', 'RET', 'CBX5', 'SNRPA1', 'FAM96A', 'TPM1', 'RPS2', 'SPG20', 'CREB1', 'SKP2', 'NUDC', 'MYD88', 'JUN', 'AGO4', 'SF3A3', 'SNRPD3', 'PYM1', 'AGO1', 'APP', 'ATG4A', 'SIAH2', 'TAF4', 'NBN', 'RBBP6', 'AGO2', 'IRS2', 'SOCS1', 'BAP1', 'BCL6', 'CARD11', 'CDKN1A', 'CEBPA', 'CNOT3', 'CXCR4', 'DDB2', 'DDIT3', 'DNMT3A', 'EIF3E', 'ERCC5', 'FANCC', 'FANCG', 'HMGA1', 'IKZF1', 'KLF4', 'MALT1', 'MDM4', 'MLLT1', 'MLLT3', 'NPM1', 'NTRK1', 'PAFAH1B2', 'PCBP1', 'PML', 'PPARG', 'PPP2R1A', 'SFPQ', 'STAT3', 'TAL1', 'TGFBR2', 'TNFAIP3', 'TSC2', 'U2AF1', 'VTI1A', 'CDC27', 'BRCA1', 'POLR3B', 'CDH1', 'CTDP1', 'CASP8', 'KARS', 'RHOA', 'UBE2D1', 'ACTB', 'MAPK9', 'MEF2A', 'CUL1', 'PSMC4', 'PRKACA', 'CDC42', 'POLD1', 'HSPA5', 'CUL3', 'UBE3C', 'POLR1A', 'RIOK2', 'RAD51', 'PABPC1', 'HDAC9', 'FBXW11', 'PSMB1', 'CSNK2A2', 'PPP2R5B', 'NOP58', 'HIPK2', 'MAP2K4', 'NEDD4', 'MCM2', 'TOP2B', 'NEDD4L', 'RAD18', 'CREBBP', 'RECQL', 'HDAC7', 'MAP2K7', 'ELAVL1', 'UPF1', 'PSMA4', 'CFTR', 'ACTN1', 'SMARCE1', 'NFKB2', 'CTNNA1', 'KDM1A', 'MARK3', 'ZRANB1', 'FYN', 'TNK2', 'UFD1L', 'GAB2', 'MARK2', 'TSG101', 'CLNS1A', 'BIRC3', 'SLC25A5', 'YBX1', 'U2AF2', 'CASP10', 'SREBF1', 'BCL3', 'TP63', 'TPR']\n",
      "\n"
     ]
    }
   ],
   "source": [
    "SLcsv(SLslorthComp,'SlorthComp')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# DAISY"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "      Gene A   Gene B\n",
      "0      ACAP1     DEF6\n",
      "1      ACAP1   GIMAP1\n",
      "2      ACAP1   MAP4K1\n",
      "3      ACAP1   SEMA4A\n",
      "4        ACD  SMARCC2\n",
      "...      ...      ...\n",
      "2811  ZNF407    MEOX2\n",
      "2812  ZC3H7B    MUTYH\n",
      "2813  ZNF593  MYBBP1A\n",
      "2814  ZC3H7B    RNF40\n",
      "2815   ZWINT     SKP2\n",
      "\n",
      "[2816 rows x 2 columns]\n"
     ]
    }
   ],
   "source": [
    "# Carga de los datos de DAISY.\n",
    "SLdaisy = pd.read_csv(f'{work_folder}DAISY/DAISY_SLpairs.csv')\n",
    "print(SLdaisy)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     n1.name  n2.name\n",
      "0      ACAP1     DEF6\n",
      "1      ACAP1   GIMAP1\n",
      "2      ACAP1   MAP4K1\n",
      "3      ACAP1   SEMA4A\n",
      "4        ACD  SMARCC2\n",
      "...      ...      ...\n",
      "2811  ZNF407    MEOX2\n",
      "2812  ZC3H7B    MUTYH\n",
      "2813  ZNF593  MYBBP1A\n",
      "2814  ZC3H7B    RNF40\n",
      "2815   ZWINT     SKP2\n",
      "\n",
      "[2816 rows x 2 columns]\n"
     ]
    }
   ],
   "source": [
    "SLdaisy = SLdaisy.rename(columns = {'Gene A':'n1.name','Gene B':'n2.name'})\n",
    "print(SLdaisy)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------SL pairs for Essential PatuT Genes in DAISY:\n",
      "\n",
      "SL pairs in DAISY for PRMT3: No SL pairs Found\n",
      "\n",
      "SL pairs in DAISY for ASXL1: No SL pairs Found\n",
      "\n",
      "SL pairs in DAISY for HELLS: ['PCNA', 'RFC2', 'SFPQ', 'TOPBP1', 'ZNF107', 'FEN1']\n",
      "\n",
      "SL pairs in DAISY for SMARCE1: No SL pairs Found\n",
      "\n",
      "SL pairs in DAISY for KDM4B: ['POGZ', 'SMARCC2']\n",
      "\n",
      "SL pairs in DAISY for EPC2: No SL pairs Found\n",
      "\n",
      "-------SL pairs for Suppressor PatuT Genes in DAISY:\n",
      "\n",
      "SL pairs in DAISY for WHSC1: ['EZH2', 'CDCA3']\n",
      "\n",
      "SL pairs in DAISY for SUV39H2: ['RAD51']\n",
      "\n",
      "-------SL pairs for Essential PatuS Genes in DAISY:\n",
      "\n",
      "SL pairs in DAISY for CHD7: No SL pairs Found\n",
      "\n",
      "SL pairs in DAISY for BRD4: ['CCNT1', 'GGA3', 'TNK2']\n",
      "\n",
      "SL pairs in DAISY for ING5: ['MRPS25']\n",
      "\n",
      "-------SL pairs for Suppressor PatuS Genes in DAISY:\n",
      "\n",
      "SL pairs in DAISY for PHF23: No SL pairs Found\n",
      "\n",
      "SL pairs in DAISY for UBE2A: No SL pairs Found\n",
      "\n"
     ]
    }
   ],
   "source": [
    "SLcsv(SLdaisy,'DAISY')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# BioGrid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_17508/204943909.py:1: DtypeWarning: Columns (1,2,18) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  Databg = pd.read_csv(f'{work_folder}BIOGRID-ORGANISM-4.4.227.tab3/BIOGRID-ORGANISM-Homo_sapiens-4.4.227.tab3.txt',sep = '\\t')\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         #BioGRID Interaction ID Entrez Gene Interactor A  \\\n",
      "0                            103                     6416   \n",
      "1                            117                    84665   \n",
      "2                            183                       90   \n",
      "3                            278                     2624   \n",
      "4                            418                     6118   \n",
      "...                          ...                      ...   \n",
      "1184828                  3585521                    22866   \n",
      "1184829                  3585522                      273   \n",
      "1184830                  3585523                    84445   \n",
      "1184831                  3585524                    23032   \n",
      "1184832                  3585525                    51562   \n",
      "\n",
      "        Entrez Gene Interactor B  BioGRID ID Interactor A  \\\n",
      "0                           2318                   112315   \n",
      "1                             88                   124185   \n",
      "2                           2339                   106605   \n",
      "3                           5371                   108894   \n",
      "4                           6774                   112038   \n",
      "...                          ...                      ...   \n",
      "1184828                     1956                   116534   \n",
      "1184829                     1956                   106770   \n",
      "1184830                     1956                   124083   \n",
      "1184831                     1956                   116671   \n",
      "1184832                     1956                   119612   \n",
      "\n",
      "         BioGRID ID Interactor B Systematic Name Interactor A  \\\n",
      "0                         108607                            -   \n",
      "1                         106603                            -   \n",
      "2                         108625                            -   \n",
      "3                         111384                            -   \n",
      "4                         112651                  RP4-547C9.3   \n",
      "...                          ...                          ...   \n",
      "1184828                   108276                 RP11-450P7.2   \n",
      "1184829                   108276                            -   \n",
      "1184830                   108276                 RP11-108L7.8   \n",
      "1184831                   108276                            -   \n",
      "1184832                   108276                       BM-015   \n",
      "\n",
      "        Systematic Name Interactor B Official Symbol Interactor A  \\\n",
      "0                                  -                       MAP2K4   \n",
      "1                                  -                         MYPN   \n",
      "2                                  -                        ACVR1   \n",
      "3                                  -                        GATA2   \n",
      "4                                  -                         RPA2   \n",
      "...                              ...                          ...   \n",
      "1184828                            -                       CNKSR2   \n",
      "1184829                            -                         AMPH   \n",
      "1184830                            -                        LZTS2   \n",
      "1184831                            -                        USP33   \n",
      "1184832                            -                         MBIP   \n",
      "\n",
      "        Official Symbol Interactor B  \\\n",
      "0                               FLNC   \n",
      "1                              ACTN2   \n",
      "2                               FNTA   \n",
      "3                                PML   \n",
      "4                              STAT3   \n",
      "...                              ...   \n",
      "1184828                         EGFR   \n",
      "1184829                         EGFR   \n",
      "1184830                         EGFR   \n",
      "1184831                         EGFR   \n",
      "1184832                         EGFR   \n",
      "\n",
      "                                     Synonyms Interactor A  ...  \\\n",
      "0        JNKK|JNKK1|MAPKK4|MEK4|MKK4|PRKMK4|SAPKK-1|SAP...  ...   \n",
      "1                                   CMD1DD|CMH22|MYOP|RCM4  ...   \n",
      "2                  ACTRI|ACVR1A|ACVRLK2|ALK2|FOP|SKR1|TSRI  ...   \n",
      "3                                 DCML|IMD21|MONOMAC|NFE1B  ...   \n",
      "4                            REPA2|RP-A p32|RP-A p34|RPA32  ...   \n",
      "...                                                    ...  ...   \n",
      "1184828                                   CNK2|KSR2|MAGUIN  ...   \n",
      "1184829                                              AMPH1  ...   \n",
      "1184830                                            LAPSER1  ...   \n",
      "1184831                                               VDU1  ...   \n",
      "1184832                                                  -  ...   \n",
      "\n",
      "        TREMBL Accessions Interactor B  \\\n",
      "0                               Q59H94   \n",
      "1                        Q59FD9|F6THM6   \n",
      "2                                    -   \n",
      "3                                    -   \n",
      "4                                    -   \n",
      "...                                ...   \n",
      "1184828                              -   \n",
      "1184829                              -   \n",
      "1184830                              -   \n",
      "1184831                              -   \n",
      "1184832                              -   \n",
      "\n",
      "                            REFSEQ Accessions Interactor B Ontology Term IDs  \\\n",
      "0                                   NP_001120959|NP_001449                 -   \n",
      "1                      NP_001094|NP_001265272|NP_001265273                 -   \n",
      "2                                                NP_002018                 -   \n",
      "3        NP_150250|NP_150253|NP_150252|NP_150247|NP_150...                 -   \n",
      "4        NP_644805|NP_003141|NP_001356447|NP_001356443|...                 -   \n",
      "...                                                    ...               ...   \n",
      "1184828  NP_001333829|NP_001333828|NP_958440|NP_005219|...                 -   \n",
      "1184829  NP_001333829|NP_001333828|NP_958440|NP_005219|...                 -   \n",
      "1184830  NP_001333829|NP_001333828|NP_958440|NP_005219|...                 -   \n",
      "1184831  NP_001333829|NP_001333828|NP_958440|NP_005219|...                 -   \n",
      "1184832  NP_001333829|NP_001333828|NP_958440|NP_005219|...                 -   \n",
      "\n",
      "        Ontology Term Names Ontology Term Categories  \\\n",
      "0                         -                        -   \n",
      "1                         -                        -   \n",
      "2                         -                        -   \n",
      "3                         -                        -   \n",
      "4                         -                        -   \n",
      "...                     ...                      ...   \n",
      "1184828                   -                        -   \n",
      "1184829                   -                        -   \n",
      "1184830                   -                        -   \n",
      "1184831                   -                        -   \n",
      "1184832                   -                        -   \n",
      "\n",
      "         Ontology Term Qualifier IDs  Ontology Term Qualifier Names  \\\n",
      "0                                  -                              -   \n",
      "1                                  -                              -   \n",
      "2                                  -                              -   \n",
      "3                                  -                              -   \n",
      "4                                  -                              -   \n",
      "...                              ...                            ...   \n",
      "1184828                            -                              -   \n",
      "1184829                            -                              -   \n",
      "1184830                            -                              -   \n",
      "1184831                            -                              -   \n",
      "1184832                            -                              -   \n",
      "\n",
      "        Ontology Term Types Organism Name Interactor A  \\\n",
      "0                         -               Homo sapiens   \n",
      "1                         -               Homo sapiens   \n",
      "2                         -               Homo sapiens   \n",
      "3                         -               Homo sapiens   \n",
      "4                         -               Homo sapiens   \n",
      "...                     ...                        ...   \n",
      "1184828                   -               Homo sapiens   \n",
      "1184829                   -               Homo sapiens   \n",
      "1184830                   -               Homo sapiens   \n",
      "1184831                   -               Homo sapiens   \n",
      "1184832                   -               Homo sapiens   \n",
      "\n",
      "        Organism Name Interactor B  \n",
      "0                     Homo sapiens  \n",
      "1                     Homo sapiens  \n",
      "2                     Homo sapiens  \n",
      "3                     Homo sapiens  \n",
      "4                     Homo sapiens  \n",
      "...                            ...  \n",
      "1184828               Homo sapiens  \n",
      "1184829               Homo sapiens  \n",
      "1184830               Homo sapiens  \n",
      "1184831               Homo sapiens  \n",
      "1184832               Homo sapiens  \n",
      "\n",
      "[1184833 rows x 37 columns]\n"
     ]
    }
   ],
   "source": [
    "# Carga de los datos de BioGRID-\n",
    "Databg = pd.read_csv(f'{work_folder}BIOGRID-ORGANISM-4.4.227.tab3/BIOGRID-ORGANISM-Homo_sapiens-4.4.227.tab3.txt',sep = '\\t')\n",
    "print(Databg)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['#BioGRID Interaction ID', 'Entrez Gene Interactor A',\n",
      "       'Entrez Gene Interactor B', 'BioGRID ID Interactor A',\n",
      "       'BioGRID ID Interactor B', 'Systematic Name Interactor A',\n",
      "       'Systematic Name Interactor B', 'Official Symbol Interactor A',\n",
      "       'Official Symbol Interactor B', 'Synonyms Interactor A',\n",
      "       'Synonyms Interactor B', 'Experimental System',\n",
      "       'Experimental System Type', 'Author', 'Publication Source',\n",
      "       'Organism ID Interactor A', 'Organism ID Interactor B', 'Throughput',\n",
      "       'Score', 'Modification', 'Qualifications', 'Tags', 'Source Database',\n",
      "       'SWISS-PROT Accessions Interactor A', 'TREMBL Accessions Interactor A',\n",
      "       'REFSEQ Accessions Interactor A', 'SWISS-PROT Accessions Interactor B',\n",
      "       'TREMBL Accessions Interactor B', 'REFSEQ Accessions Interactor B',\n",
      "       'Ontology Term IDs', 'Ontology Term Names', 'Ontology Term Categories',\n",
      "       'Ontology Term Qualifier IDs', 'Ontology Term Qualifier Names',\n",
      "       'Ontology Term Types', 'Organism Name Interactor A',\n",
      "       'Organism Name Interactor B'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "print(Databg.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "        Official Symbol Interactor A Official Symbol Interactor B  \\\n",
      "0                             MAP2K4                         FLNC   \n",
      "1                               MYPN                        ACTN2   \n",
      "2                              ACVR1                         FNTA   \n",
      "3                              GATA2                          PML   \n",
      "4                               RPA2                        STAT3   \n",
      "...                              ...                          ...   \n",
      "1184828                       CNKSR2                         EGFR   \n",
      "1184829                         AMPH                         EGFR   \n",
      "1184830                        LZTS2                         EGFR   \n",
      "1184831                        USP33                         EGFR   \n",
      "1184832                         MBIP                         EGFR   \n",
      "\n",
      "              Experimental System  \n",
      "0                      Two-hybrid  \n",
      "1                      Two-hybrid  \n",
      "2                      Two-hybrid  \n",
      "3                      Two-hybrid  \n",
      "4                      Two-hybrid  \n",
      "...                           ...  \n",
      "1184828  Affinity Capture-Western  \n",
      "1184829  Affinity Capture-Western  \n",
      "1184830  Affinity Capture-Western  \n",
      "1184831  Affinity Capture-Western  \n",
      "1184832  Affinity Capture-Western  \n",
      "\n",
      "[1184833 rows x 3 columns]\n"
     ]
    }
   ],
   "source": [
    "# Filtrado de las columnas que contienen las parejas de la interacción y la columna que contiene el tipo de interacción.\n",
    "Databg = Databg[['Official Symbol Interactor A','Official Symbol Interactor B','Experimental System']]\n",
    "print(Databg)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Two-hybrid', 'Affinity Capture-Luminescence', 'Affinity Capture-Western', 'Reconstituted Complex', 'Biochemical Activity', 'FRET', 'Dosage Rescue', 'Co-purification', 'Protein-peptide', 'Co-localization', 'Affinity Capture-MS', 'Affinity Capture-RNA', 'Co-crystal Structure', 'Far Western', 'Phenotypic Enhancement', 'Phenotypic Suppression', 'Co-fractionation', 'Protein-RNA', 'Synthetic Rescue', 'Synthetic Lethality', 'Synthetic Growth Defect', 'PCA', 'Dosage Lethality', 'Negative Genetic', 'Dosage Growth Defect', 'Proximity Label-MS', 'Positive Genetic']\n"
     ]
    }
   ],
   "source": [
    "# Obtención de una lista con los tipos de interacción disponibles en BioGRID.\n",
    "Exp_System = Databg['Experimental System'].astype('category')\n",
    "Exp_System = Exp_System.unique().tolist()\n",
    "print(Exp_System)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     n1.name  n2.name\n",
      "0      SEC14     SNCA\n",
      "1       CKI1     SNCA\n",
      "2      FBXW7    BUB1B\n",
      "3      FBXW7   FANCD2\n",
      "4      FBXW7    RAB4B\n",
      "...      ...      ...\n",
      "2162    TP53    AP3B2\n",
      "2163    TP53   ANXA11\n",
      "2164     ME2      ME3\n",
      "2165    LRP8  SLC31A1\n",
      "2166    LRP8  SLC7A11\n",
      "\n",
      "[2167 rows x 2 columns]\n"
     ]
    }
   ],
   "source": [
    "# Obtención de las interacciones de BioGRID correspondientes a SL.\n",
    "SLbg = (((Databg[Databg['Experimental System'] == 'Synthetic Lethality']).reset_index(drop = True))\\\n",
    "        [['Official Symbol Interactor A', 'Official Symbol Interactor B']]).rename\\\n",
    "            (columns = {'Official Symbol Interactor A':'n1.name', 'Official Symbol Interactor B':'n2.name'})\n",
    "print(SLbg)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------SL pairs for Essential PatuT Genes in BioGrid:\n",
      "\n",
      "SL pairs in BioGrid for PRMT3: No SL pairs Found\n",
      "\n",
      "SL pairs in BioGrid for ASXL1: No SL pairs Found\n",
      "\n",
      "SL pairs in BioGrid for HELLS: No SL pairs Found\n",
      "\n",
      "SL pairs in BioGrid for SMARCE1: No SL pairs Found\n",
      "\n",
      "SL pairs in BioGrid for KDM4B: No SL pairs Found\n",
      "\n",
      "SL pairs in BioGrid for EPC2: No SL pairs Found\n",
      "\n",
      "-------SL pairs for Suppressor PatuT Genes in BioGrid:\n",
      "\n",
      "SL pairs in BioGrid for WHSC1: No SL pairs Found\n",
      "\n",
      "SL pairs in BioGrid for SUV39H2: ['KRAS']\n",
      "\n",
      "-------SL pairs for Essential PatuS Genes in BioGrid:\n",
      "\n",
      "SL pairs in BioGrid for CHD7: No SL pairs Found\n",
      "\n",
      "SL pairs in BioGrid for BRD4: ['PTAR1', 'CHEK2', 'CHEK1', 'WEE1', 'CDC7', 'CDK9']\n",
      "\n",
      "SL pairs in BioGrid for ING5: ['MAP2K1', 'CDC7', 'CDK9', 'TUBA1A']\n",
      "\n",
      "-------SL pairs for Suppressor PatuS Genes in BioGrid:\n",
      "\n",
      "SL pairs in BioGrid for PHF23: No SL pairs Found\n",
      "\n",
      "SL pairs in BioGrid for UBE2A: No SL pairs Found\n",
      "\n"
     ]
    }
   ],
   "source": [
    "SLcsv(SLbg,'BioGrid')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# GenomeRNAi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_17508/2445209705.py:1: DtypeWarning: Columns (2,5,7) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  DataGRNAi = pd.read_csv(f'{work_folder}GenomeRNAi_v17_AllScreens/GenomeRNAi_v17_Homo_sapiens_clean.txt',sep = '\\t')\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "          #Stable ID Entrez ID Gene ID Gene Symbol  Reagent ID  \\\n",
      "0          GR00016-A    113510     NaN      HEL308          np   \n",
      "1          GR00016-A     57531   57531       HACE1          np   \n",
      "2          GR00016-A      9493    9493       KIF23          np   \n",
      "3          GR00016-A     57410   57410       SCYL1          np   \n",
      "4          GR00016-A     83858   83858      ATAD3B          np   \n",
      "...              ...       ...     ...         ...         ...   \n",
      "2416177  GR00402-S-2     85478   85478    NYD-SP28  SIR0014272   \n",
      "2416178  GR00402-S-2      9402    9402       GRAP2  SIR0003731   \n",
      "2416179  GR00402-S-2      4089    4089       SMAD4  SIR0001447   \n",
      "2416180  GR00402-S-2    116337  116337       PANX3  SIR0014648   \n",
      "2416181           //       NaN     NaN         NaN         NaN   \n",
      "\n",
      "                        Score  \\\n",
      "0                        2.86   \n",
      "1                        2.54   \n",
      "2                        2.04   \n",
      "3                        1.54   \n",
      "4                        1.74   \n",
      "...                       ...   \n",
      "2416177  -0.40697235557445227   \n",
      "2416178     1.075584431713187   \n",
      "2416179   -0.3379224332027875   \n",
      "2416180   0.18747896419237745   \n",
      "2416181                   NaN   \n",
      "\n",
      "                                                 Phenotype Conditions  \\\n",
      "0        Upregulation of Wnt/beta-catenin pathway after...        NaN   \n",
      "1        Upregulation of Wnt/beta-catenin pathway after...        NaN   \n",
      "2        Upregulation of Wnt/beta-catenin pathway after...        NaN   \n",
      "3                                                     none        NaN   \n",
      "4                                                     none        NaN   \n",
      "...                                                    ...        ...   \n",
      "2416177                                          no effect        NaN   \n",
      "2416178                                          no effect        NaN   \n",
      "2416179                                          no effect        NaN   \n",
      "2416180                                          no effect        NaN   \n",
      "2416181                                                NaN        NaN   \n",
      "\n",
      "        Follow Up                Comment  \n",
      "0              no                    NaN  \n",
      "1              no                    NaN  \n",
      "2              no                    NaN  \n",
      "3              no                    NaN  \n",
      "4              no                    NaN  \n",
      "...           ...                    ...  \n",
      "2416177       NaN  5 days post-infection  \n",
      "2416178       NaN  5 days post-infection  \n",
      "2416179       NaN  5 days post-infection  \n",
      "2416180       NaN  5 days post-infection  \n",
      "2416181       NaN                    NaN  \n",
      "\n",
      "[2416182 rows x 10 columns]\n"
     ]
    }
   ],
   "source": [
    "# Carga de los datos de GenomeRNAi\n",
    "DataGRNAi = pd.read_csv(f'{work_folder}GenomeRNAi_v17_AllScreens/GenomeRNAi_v17_Homo_sapiens_clean.txt',sep = '\\t')\n",
    "print(DataGRNAi)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "        Gene Symbol                                          Phenotype\n",
      "0            HEL308  Upregulation of Wnt/beta-catenin pathway after...\n",
      "1             HACE1  Upregulation of Wnt/beta-catenin pathway after...\n",
      "2             KIF23  Upregulation of Wnt/beta-catenin pathway after...\n",
      "3             SCYL1                                               none\n",
      "4            ATAD3B                                               none\n",
      "...             ...                                                ...\n",
      "2416177    NYD-SP28                                          no effect\n",
      "2416178       GRAP2                                          no effect\n",
      "2416179       SMAD4                                          no effect\n",
      "2416180       PANX3                                          no effect\n",
      "2416181         NaN                                                NaN\n",
      "\n",
      "[2416182 rows x 2 columns]\n"
     ]
    }
   ],
   "source": [
    "# Filtrado de las columnas que contienen el gen y el fenotipo inferido para cada uno como resultado de su screening.\n",
    "DataGRNAi = DataGRNAi[['Gene Symbol','Phenotype']]\n",
    "print(DataGRNAi)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Upregulation of Wnt/beta-catenin pathway after WNT3A stimulation\n",
      "none\n",
      "nan\n",
      "Phenotype\n",
      "Wnt reporter downregulated\n",
      "Wnt reporter upregulated\n",
      "Synthetic lethal with Ras\n",
      "Increased gamma-H2AX phosphorylation\n",
      "Decreased viability with paclitaxel\n",
      "Increased cell migration\n",
      "Decreased cell migration\n",
      "Decreased viability\n",
      "Decreased melanin production\n",
      "Upregulation of Wnt pathway\n",
      "Downregulation of Wnt pathway after Wnt3A stimulation\n",
      "Upregulation of Wnt pathway after Wnt3A stimulation\n",
      "Increased apoptosis\n",
      "Apoptosis resistance\n",
      "Increased proliferation\n",
      "Colony formation\n",
      "Decreased nuclei size in G2M\n",
      "Increased cell number in G2M, increased nuclei size in G2M\n",
      "Increased cell number in G1, average or increased nuclei size in G1\n",
      "Increased cell number in G1, small nuclei in G1\n",
      "Increased cell number in G2M, increased number of polyploid cells, increased number of cells with high perimeter-to-area ratio nuclei\n",
      "Increased cell number in G2M, increased number of polyploid cells\n",
      "Increased cell number in S and G2M\n",
      "Increased cell number in S\n",
      "Mitotic arrest, cytokinesis defect\n",
      "Mitotic arrest\n",
      "Mitotic arrest, spindle defect\n",
      "Cytokinesis defect\n",
      "Lethal upon entry into mitosis\n",
      "Lethal upon entry into mitosis, spindle defect\n",
      "Cell attachment phenotype\n",
      "Mitotic arrest, cytokinesis defect, spindle defect\n",
      "Increased cell growth\n",
      "Increased 8N DNA content\n",
      "Decreased cell number, increased S DNA content\n",
      "Increased G2M DNA content\n",
      "Increased G1 DNA content\n",
      "Decreased cell number, increased G1 DNA content\n",
      "Increased G2M DNA content, increased 8N DNA content\n",
      "Increased S DNA content\n",
      "Decreased cell number, increased G2M DNA content, increased 8N DNA content\n",
      "Decreased cell number\n",
      "Increased S DNA content, increased G2M DNA content\n",
      "Decreased cell number, increased G2M DNA content\n",
      "Decreased cell number, increased S DNA content, increased G2M DNA content\n",
      "Increased G1 DNA content, increased S DNA content\n",
      "Decreased cell number, increased S DNA content, increased G2M DNA content, increased 8N DNA content\n",
      "Increased S DNA content, increased 8N DNA content\n",
      "Decreased cell number, increased 8N DNA content\n",
      "Decreased cell number, increased S DNA content, increased 8N DNA content\n",
      "Increased S DNA content, increased G2M DNA content, increased 8N DNA content\n",
      "Cell division defect\n",
      "G0/1 arrest\n",
      "S arrest\n",
      "G2 arrest\n",
      "Increased number of mitotic cells\n",
      "Increased cell size\n",
      "Mitotic spindle defects\n",
      "Increased mitotic index\n",
      "Synthetic lethal with cisplatin\n",
      "Synthetic lethal with gemcitabine\n",
      "Synthetic lethal with paclitaxel\n",
      "Decreased viability with cisplatin\n",
      "Essential for multiple myeloma proliferation\n",
      "Increased cell death HMECs cells\n",
      "Increased cell death in HCT116 cells\n",
      "Increased cell death in HCC-1954 cells\n",
      "Increased cell death in DLD-1 cells\n",
      "Increased cell death in breast cancer cell lines (MCF10A, MDA-MB-435)\n",
      "Decreased viability in CMK cells\n",
      "Decreased viability in HMC1.1 cells\n",
      "Decreased viability after gemcitabine stimulation\n",
      "Decreased mitotic index\n",
      "Increased NF-kappaB reporter activity\n",
      "Increased sensitivity to paclitaxel\n",
      "Decreased sensitivity to paclitaxel\n",
      "Paclitaxel antagonistic effect\n",
      "Paclitaxel sensitizing effect\n",
      "Decreased viability with poly (ADP‐ribose)‐polymerase‐1 (PARP) inhibitor\n",
      "Increased HDAC inhibitor resistance\n",
      "Increased resistance to Trastuzumab\n",
      "Increased cell transformation\n",
      "Increased viability with tamoxifen\n",
      "Decreased viability in Von Hippel-Lindau (VHL) (-/-) cell line\n",
      "Reduced beta-catenin activity and colon cancer cell proliferation\n",
      "Increased viability with nutlin-3\n",
      "Decreased Salmonella enterica Typhimurium invasion\n",
      "Increased Salmonella enterica Typhimurium invasion\n",
      "Decreased Salmonella-containing vacuole maturation\n",
      "Increased Salmonella-containing vacuole maturation\n",
      "Decreased Salmonella enterica Typhimurium binding\n",
      "Increased Salmonella enterica Typhimurium binding\n",
      "Decreased Salmonella enterica Typhimurium effector injection\n",
      "Increased Salmonella enterica Typhimurium effector injection\n",
      "Decreased Salmonella enterica Typhimurium ruffling\n",
      "Increased Salmonella enterica Typhimurium ruffling\n",
      "Decreased Salmonella enterica Typhimurium membrane closure\n",
      "Increased Salmonella enterica Typhimurium membrane closure\n",
      "Downregulation of TNF-α/NF-κB pathway\n",
      "Decreased CXCL12 induced migration\n",
      "Increased CXCL12 induced migration\n",
      "Decreased 12E8 tau but not total tau protein expression and decreased ratio of 12E8/total tau protein expression\n",
      "Decreased 12E8 and total tau protein expression\n",
      "Decreased influenza A virus infection\n",
      "Increased influenza A virus infection\n",
      "Increased cilium length after serum starvation\n",
      "Decreased cilium length after serum starvation\n",
      "Increased cilium length\n",
      "Decreased cilium length\n",
      "Decreased homologous recombination repair frequency\n",
      "Increased homologous recombination repair frequency\n",
      "Increased viability with TRAIL\n",
      "Decreased viability with TRAIL\n",
      "Decreased CYP1A1 activity after TCDD stimulation\n",
      "Decreased telomerase activity\n",
      "Increased telomerase activity\n",
      "Decreased viability with RAF265\n",
      "Decreased p24 protein expression\n",
      "Decreased Tat-dependent transcription\n",
      "Decreased cell number and increased metaphase cell number\n",
      "High actin ratio cells\n",
      "Metaphase cells\n",
      "Actin fiber cells\n",
      "Big cells\n",
      "Large cells\n",
      "Lamellipodia cells\n",
      "Lamellipodia and high actin ratio cells\n",
      "Proliferating cells\n",
      "Bright nuclei\n",
      "Large nuclei\n",
      "Cells with protrusions\n",
      "Elongated cells\n",
      "Elongated cells with protrusions\n",
      "Abundance of large cells with protrusions and bright nuclei\n",
      "Small cells\n",
      "Low eccentricity cells\n",
      "Increased viability with paclitaxel\n",
      "Decreased Hepatitis C virus replication\n",
      "Increased Hepatitis C virus replication\n",
      "Synthetic lethal with imatinib mesylate\n",
      "Decreased POU5F1-GFP protein expression\n",
      "Decreased OCT4 protein expression\n",
      "Decreased NANOG protein expression\n",
      "Increased release from monastrol-induced mitotic arrest\n",
      "Increased viability after borna disease (rVSVΔG*/BDVG) virus infection\n",
      "Decreased substrate adherent cell growth\n",
      "Decreased influenza A/WSN/33 replication\n",
      "Decreased influenza A H1N1 (A/WSN/33) virus numbers\n",
      "Decreased influenza A H1N1 (A/Hamburg/04/2009) virus numbers\n",
      "Decreased TP53 protein expression ratio (wild-type / TP53 knockout cells)\n",
      "Decreased viability of wild-type and TP53 knockout cells\n",
      "Decreased viability of wild-type and TP53 knockout cells, decreased TP53 protein expression ratio (wild-type / TP53 knockout cells)\n",
      "Decreased viability of wild-type and TP53 knockout cells, increased TP53 protein expression ratio (wild-type / TP53 knockout cells)\n",
      "Increased TP53 protein expression ratio (wild-type / TP53 knockout cells)\n",
      "Increased HPV18 LCR reporter activity\n",
      "Decreased BPV1 E2 protein expression\n",
      "Increased number of cells in monopolar arrest with EMD534085 (a Kinesin-5 inhibitor)\n",
      "Decreased number of cells in monopolar arrest with EMD534085 (a Kinesin-5 inhibitor)\n",
      "Downregulation of NF-kappaB pathway after LMP1 stimulation\n",
      "Downregulation of NF-kappaB pathway after IL-1beta stimulation\n",
      "Downregulation of NF-kappaB pathway after TNFalpha stimulation\n",
      "Upregulation of NF-kappaB pathway after LMP1 stimulation\n",
      "Increased viability with 4OH tamoxifen\n",
      "Decreased G3BP1 protein expression and assembly in stress granules after arsenite stimulation\n",
      "Decreased DCP1a protein expression and assembly in processing bodies after arsenite stimulation\n",
      "Decreased G3BP1 protein expression and assembly in stress granules and decreased DCP1a protein expression and assembly in processing bodies after arsenite stimulation\n",
      "Increased Erk phosphorylation after EGF stimulation\n",
      "Increased Akt phosphorylation after EGF stimulation\n",
      "Increased mitotic exit time\n",
      "Decreased viability after Chlamydia trachomatis serovar L2 infection and TNF-alpha/CHX stimulation\n",
      "Decreased viability with carboplatin\n",
      "Nucleolar pre-40S maturation defects\n",
      "Nuclear pre-40S maturation defects\n",
      "Nucleoplasmic pre-40S maturation defects\n",
      "Cytoplasmic 40S maturation defects\n",
      "Nuclear 40S maturation defects\n",
      "Nuclear 60S biogenesis defects\n",
      "Altered focal adhesion (FA) or cell shape morphology\n",
      "Decreased focal adhesion (FA) area, decreased FA length, decreased FA mean intensity, increased number of small and round FAs\n",
      "Decreased focal adhesion (FA) area, decreased FA length, decreased FA mean intensity, increased number of small and round FAs, decreased cell elongation, decreased cell spreading\n",
      "Decreased focal adhesion (FA) area, decreased FA length, decreased FA mean intensity, increased number of small and round FAs, decreased cell spreading\n",
      "Decreased focal adhesion (FA) area, decreased FA length, decreased FA mean intensity, increased number of small and round FAs, decreased FA abundance\n",
      "Decreased cell spreading\n",
      "Decreased focal adhesion (FA) area, decreased FA length, decreased FA mean intensity, increased number of small and round FAs, decreased FA abundance, decreased cell spreading\n",
      "Decreased focal adhesion (FA) abundance\n",
      "Decreased focal adhesion (FA) area, decreased FA length, decreased FA mean intensity, increased number of small and round FAs, decreased FA abundance, decreased peripheral FA formation\n",
      "Decreased focal adhesion (FA) area, decreased FA length, decreased FA mean intensity, increased number of small and round FAs, decreased FA abundance, decreased peripheral FA formation, decreased cell spreading\n",
      "Decreased focal adhesion (FA) area, decreased FA length, decreased FA mean intensity, increased number of small and round FAs, decreased FA abundance, decreased peripheral FA formation, increased cell spreading\n",
      "Decreased focal adhesion (FA) area, decreased FA length, decreased FA mean intensity, increased number of small and round FAs, decreased FA abundance, increased cell elongation\n",
      "Decreased focal adhesion (FA) area, decreased FA length, decreased FA mean intensity, increased number of small and round FAs, decreased FA abundance, increased cell spreading\n",
      "Decreased focal adhesion (FA) area, decreased FA length, decreased FA mean intensity, increased number of small and round FAs, decreased FA abundance, increased peripheral FA formation\n",
      "Decreased focal adhesion (FA) abundance, decreased cell spreading\n",
      "Decreased focal adhesion (FA) area, decreased FA length, decreased FA mean intensity, increased number of small and round FAs, decreased FA abundance, increased peripheral FA formation, decreased cell spreading\n",
      "Decreased focal adhesion (FA) area, decreased FA length, decreased FA mean intensity, increased number of small and round FAs, decreased FA abundance, increased peripheral FA formation, decreased cell elongation, decreased cell spreading\n",
      "Decreased focal adhesion (FA) area, decreased FA length, decreased FA mean intensity, increased number of small and round FAs, decreased FA abundance, increased peripheral FA formation, decreased cell elongation\n",
      "Decreased focal adhesion (FA) abundance, increased cell elongation\n",
      "Decreased focal adhesion (FA) area, decreased FA length, decreased FA mean intensity, increased number of small and round FAs, decreased FA abundance, increased peripheral FA formation, increased cell spreading\n",
      "Decreased focal adhesion (FA) area, decreased FA length, decreased FA mean intensity, increased number of small and round FAs, decreased FA abundance, increased peripheral FA formation, increased cell elongation\n",
      "Decreased focal adhesion (FA) area, decreased FA length, decreased FA mean intensity, increased number of small and round FAs, decreased FA abundance, increased peripheral FA formation, increased cell elongation, decreased cell spreading\n",
      "Decreased focal adhesion (FA) abundance, increased cell elongation, decreased cell spreading\n",
      "Decreased focal adhesion (FA) area, decreased FA length, decreased FA mean intensity, increased number of small and round FAs, increased cell elongation\n",
      "Decreased focal adhesion (FA) abundance, increased cell spreading\n",
      "Decreased focal adhesion (FA) abundance, increased peripheral FA formation\n",
      "Decreased focal adhesion (FA) area, decreased FA length, decreased FA mean intensity, increased number of small and round FAs, increased cell spreading\n",
      "Decreased focal adhesion (FA) area, decreased FA length, decreased FA mean intensity, increased number of small and round FAs, increased FA abundance\n",
      "Decreased focal adhesion (FA) area, decreased FA length, decreased FA mean intensity, increased number of small and round FAs, increased FA abundance, decreased cell spreading\n",
      "Decreased focal adhesion (FA) area, decreased FA length, decreased FA mean intensity, increased number of small and round FAs, increased FA abundance, increased cell spreading\n",
      "Decreased focal adhesion (FA) area, decreased FA length, decreased FA mean intensity, increased number of small and round FAs, increased peripheral FA formation\n",
      "Decreased focal adhesion (FA) area, decreased FA length, decreased FA mean intensity, increased number of small and round FAs, increased peripheral FA formation, decreased cell spreading\n",
      "Decreased focal adhesion (FA) area, decreased FA length, decreased FA mean intensity, increased number of small and round FAs, increased peripheral FA formation, increased cell elongation\n",
      "Decreased focal adhesion (FA) area, decreased FA length, decreased FA mean intensity, increased number of small and round FAs, increased peripheral FA formation, increased cell spreading\n",
      "Decreased focal adhesion (FA) area, decreased FA length, increased number of small and round FAs\n",
      "Decreased focal adhesion (FA) area, decreased FA length, increased number of small and round FAs, decreased cell spreading\n",
      "Decreased focal adhesion (FA) area, decreased FA length, increased number of small and round FAs, decreased FA abundance, increased peripheral FA formation, increased cell elongation\n",
      "Decreased focal adhesion (FA) area, decreased FA length, increased number of small and round FAs, increased cell elongation\n",
      "Decreased focal adhesion (FA) area, decreased FA length, increased number of small and round FAs, increased peripheral FA formation, increased cell elongation\n",
      "Decreased focal adhesion (FA) area, decreased FA length, increased number of small and round FAs, increased peripheral FA formation, increased cell spreading\n",
      "Increased cell elongation\n",
      "Increased cell elongation, decreased cell spreading\n",
      "Increased cell spreading\n",
      "Increased focal adhesion (FA) abundance\n",
      "Increased focal adhesion (FA) abundance, increased cell spreading\n",
      "Increased focal adhesion (FA) area, decreased number of small and round FAs\n",
      "Increased focal adhesion (FA) area, decreased number of small and round FAs, decreased FA abundance, decreased cell spreading\n",
      "Increased focal adhesion (FA) area, decreased number of small and round FAs, decreased FA abundance, increased peripheral FA formation\n",
      "Increased focal adhesion (FA) area, decreased number of small and round FAs, decreased FA abundance, increased peripheral FA formation, decreased cell spreading\n",
      "Decreased focal adhesion (FA) abundance, increased peripheral FA formation, decreased cell elongation\n",
      "Increased focal adhesion (FA) area, decreased number of small and round FAs, increased peripheral FA formation\n",
      "Increased focal adhesion (FA) area, decreased number of small and round FAs, increased peripheral FA formation, increased cell spreading\n",
      "Increased number of small and round focal adhesions (FAs)\n",
      "Increased number of small and round focal adhesions (FAs), decreased FA abundance\n",
      "Increased number of small and round focal adhesions (FAs), decreased FA abundance, decreased cell spreading\n",
      "Increased number of small and round focal adhesions (FAs), decreased FA abundance, increased peripheral FA formation\n",
      "Increased number of small and round focal adhesions (FAs), decreased FA abundance, increased peripheral FA formation, increased cell elongation\n",
      "Increased number of small and round focal adhesions (FAs), decreased FA abundance, increased peripheral FA formation, increased cell spreading\n",
      "Increased peripheral focal adhesion (FA) formation\n",
      "Increased peripheral focal adhesion (FA) formation, increased cell elongation, decreased cell spreading\n",
      "Decreased focal adhesion (FA) abundance, increased peripheral FA formation, decreased cell spreading\n",
      "Decreased focal adhesion (FA) abundance, increased peripheral FA formation, increased cell elongation\n",
      "Decreased focal adhesion (FA) abundance, increased peripheral FA formation, increased cell elongation, decreased cell spreading\n",
      "Decreased focal adhesion (FA) abundance, increased peripheral FA formation, increased cell spreading\n",
      "Increased ID2::GFP protein expression\n",
      "Increased colony dispersion (increased number of colonies and decreased number of cells per colony)\n",
      "Increased circadian rhythm amplitude\n",
      "Increased circadian period length\n",
      "Decreased circadian period length\n",
      "Synthetic lethal with c-Myc after tamoxifen stimulation\n",
      "Increased IL-10 protein expression\n",
      "Decreased IL-10 protein expression\n",
      "Increased IL-10 protein expression, increased IFN-gamma protein expression\n",
      "Decreased IL-13 protein expression\n",
      "Increased IL-13 protein expression\n",
      "Decreased IFN-gamma protein expression\n",
      "Increased IFN-gamma protein expression\n",
      "Multipolar spindles\n",
      "Acentrosomal spindle poles\n",
      "Inhibition of centrosomal clustering\n",
      "Decreased HIV-LTR-beta-galactosidase protein expression\n",
      "Increased HIV-LTR-beta-galactosidase protein expression\n",
      "Decreased and increased HIV-LTR-beta-galactosidase protein expression\n",
      "Decreased Tat-dependent HIV-LTR-beta-galactosidase protein expression\n",
      "Decreased HIV-1 infection\n",
      "Decreased viability after TNF/CHX stimulation\n",
      "Increased viability after TNF/CHX stimulation\n",
      "Increased cell viability after pRB stimulation\n",
      "Decreased cell viability after pRB stimulation\n",
      "Increased senescence-associated beta-galactosidase protein expression after pRB stimulation\n",
      "Decreased senescence-associated beta-galactosidase protein expression after pRB stimulation\n",
      "Increased viability\n",
      "Increased ionizing radiation sensitivity\n",
      "Decreased ionizing radiation sensitivity\n",
      "Decreased viability after ionizing radiation\n",
      "Decreased JFH-1 genotype 2a Hepatitis C virus (HCV) infection\n",
      "Increased JFH-1 genotype 2a Hepatitis C virus (HCV) infection\n",
      "Decreased JFH-1 genotype 2a Hepatitis C virus (HCV) infection after direct virus infection\n",
      "Increased JFH-1 genotype 2a Hepatitis C virus (HCV) infection after direct virus infection\n",
      "Decreased JFH-1 genotype 2a Hepatitis C virus (HCV) infection after viral supernatant infection\n",
      "Increased JFH-1 genotype 2a Hepatitis C virus (HCV) infection after viral supernatant infection\n",
      "Decreased Hepatitis C Virus pseudoparticles (HCVpp; H77; genotype 1a) infection\n",
      "Decreased cell–culture–derived Hepatitis C virus (HCVcc; Luc–Jc1) infection\n",
      "Decreased viability in colon lineage\n",
      "Decreased viability in esophageal squamous lineage\n",
      "Decreased viability in esophageal squamous lineage; decreased viability in glioblastoma multiforme (GBM) lineage\n",
      "Decreased viability in esophageal squamous lineage; decreased viability in glioblastoma multiforme (GBM) lineage; decreased viability in non-small-cell lung cancer (NSCLC) lineage\n",
      "Decreased viability in esophageal squamous lineage; decreased viability in glioblastoma multiforme (GBM) lineage; decreased viability in ovarian lineage\n",
      "Decreased viability in esophageal squamous lineage; decreased viability in glioblastoma multiforme (GBM) lineage; decreased viability in pancreas lineage\n",
      "Decreased viability in esophageal squamous lineage; decreased viability in non-small-cell lung cancer (NSCLC) lineage\n",
      "Decreased viability in esophageal squamous lineage; decreased viability in non-small-cell lung cancer (NSCLC) lineage; decreased viability in ovarian lineage\n",
      "Decreased viability in esophageal squamous lineage; decreased viability in ovarian lineage\n",
      "Decreased viability in esophageal squamous lineage; decreased viability in pancreas lineage\n",
      "Decreased viability in glioblastoma multiforme (GBM) lineage\n",
      "Decreased viability in glioblastoma multiforme (GBM) lineage; decreased viability in non-small-cell lung cancer (NSCLC) lineage\n",
      "Decreased viability in glioblastoma multiforme (GBM) lineage; decreased viability in ovarian lineage\n",
      "Decreased viability in glioblastoma multiforme (GBM) lineage; decreased viability in pancreas lineage\n",
      "Decreased viability in non-small-cell lung cancer (NSCLC) lineage\n",
      "Decreased viability in non-small-cell lung cancer (NSCLC) lineage; decreased viability in ovarian lineage\n",
      "Decreased viability in non-small-cell lung cancer (NSCLC) lineage; decreased viability in ovarian lineage; decreased viability in pancreas lineage\n",
      "Decreased viability in non-small-cell lung cancer (NSCLC) lineage; decreased viability in pancreas lineage\n",
      "Decreased viability in ovarian lineage\n",
      "Decreased viability in ovarian lineage; decreased viability in pancreas lineage\n",
      "Decreased viability in pancreas lineage\n",
      "Decreased viability in colon lineage; decreased viability in esophageal squamous lineage\n",
      "Decreased viability in colon lineage; decreased viability in esophageal squamous lineage; decreased viability in pancreas lineage\n",
      "Decreased viability in colon lineage; decreased viability in glioblastoma multiforme (GBM) lineage\n",
      "Decreased viability in colon lineage; decreased viability in non-small-cell lung cancer (NSCLC) lineage\n",
      "Decreased viability in colon lineage; decreased viability in non-small-cell lung cancer (NSCLC) lineage; decreased viability in pancreas lineage\n",
      "Decreased viability in colon lineage; decreased viability in ovarian lineage\n",
      "Decreased viability in colon lineage; decreased viability in pancreas lineage\n",
      "Increased G1 length\n",
      "Increased G1 length, increased S length\n",
      "Increased S length\n",
      "Increased G1 length, increased G2 length\n",
      "Increased S length, increased G2 length\n",
      "Increased G1 length, increased S length, increased G2 length\n",
      "Increased G2 length\n",
      "Decreased vesicular stomatitis virus (VSV) infection\n",
      "Decreased lymphocytic choriomeningitis virus (LCMV) infection\n",
      "Decreased human parainfluenza virus type 3 (HPIV3) infection\n",
      "Decreased Sindbis virus (SIN) capsid and autophagosome LC3 protein colocalization\n",
      "Decreased viability after sindbis virus (SIN) dsTE12Q infection\n",
      "Decreased mitophagy mCherry-Parkin protein expression after carbonyl cyanide m-chlorphenylhydrazone (CCCP) stimulation\n",
      "Decreased Cdk9 T-loop phosphorylation\n",
      "Decreased viability with neratinib\n",
      "Increased viability with neratinib\n",
      "Increased telomere dysfunction-induced foci (TIF) formation\n",
      "FOXO1 nuclear localization\n",
      "HIV Rev nuclear localization\n",
      "Decreased human cytomegalovirus (HCMV) strain AD169 replication\n",
      "Increased human cytomegalovirus (HCMV) strain AD169 replication\n",
      "Increased vaccinia virus (VACV) infection\n",
      "Decreased vaccinia virus (VACV) infection\n",
      "Synthetic lethal with MLN4924 (a NAE inhibitor)\n",
      "Increased viability with MLN4924 (a NAE inhibitor)\n",
      "Decreased shRNA abundance\n",
      "Decreased viability after Maraba virus infection\n",
      "Decreased hepcidin::fluc mRNA expression\n",
      "Increased hepcidin::fluc mRNA expression\n",
      "Increased SMN2 exon 7 inclusion\n",
      "Negative genetic interaction between BLM-/- and BLM+/+\n",
      "Negative genetic interaction between MUS81-/- and MUS81+/+\n",
      "Negative genetic interaction between PTEN-/- and PTEN+/+\n",
      "Negative genetic interaction between PTTG1-/- and PTTG1+/+\n",
      "Negative genetic interaction between KRASG13D/+ and KRAS+/-\n",
      "Misorientated spindle\n",
      "Decreased viability (cell death)\n",
      "Dynamic nuclei (hole, folded or small irregular)\n",
      "Dynamic nuclei (hole, folded or small irregular), decreased viability (cell death)\n",
      "Dynamic nuclei (hole, folded or small irregular), decreased viability (cell death), increased nuclear migration speed\n",
      "Dynamic nuclei (hole, folded or small irregular), increased nuclear migration speed\n",
      "Increased mitotic delay (prometaphase or metaphase alignment problems, MAP)\n",
      "Increased mitotic delay (prometaphase or metaphase alignment problems, MAP), decreased viability (cell death)\n",
      "Increased mitotic delay (prometaphase or metaphase alignment problems, MAP), decreased viability (cell death), increased nuclear migration speed\n",
      "Increased mitotic delay (prometaphase or metaphase alignment problems, MAP), dynamic nuclei (hole, folded or small irregular)\n",
      "Increased mitotic delay (prometaphase or metaphase alignment problems, MAP), dynamic nuclei (hole, folded or small irregular), decreased viability (cell death)\n",
      "Increased mitotic delay (prometaphase or metaphase alignment problems, MAP), large nuclei\n",
      "Increased mitotic delay (prometaphase or metaphase alignment problems, MAP), large nuclei, decreased viability (cell death)\n",
      "Increased mitotic delay (prometaphase or metaphase alignment problems, MAP), many micronuclei (grape)\n",
      "Increased mitotic delay (prometaphase or metaphase alignment problems, MAP), many micronuclei (grape), decreased viability (cell death)\n",
      "Increased mitotic delay (prometaphase or metaphase alignment problems, MAP), many micronuclei (grape), dynamic nuclei (hole, folded or small irregular)\n",
      "Increased mitotic delay (prometaphase or metaphase alignment problems, MAP), many micronuclei (grape), dynamic nuclei (hole, folded or small irregular), decreased viability (cell death)\n",
      "Increased mitotic delay (prometaphase or metaphase alignment problems, MAP), many micronuclei (grape), large nuclei\n",
      "Increased mitotic delay (prometaphase or metaphase alignment problems, MAP), many micronuclei (grape), large nuclei, decreased viability (cell death)\n",
      "Increased mitotic delay (prometaphase or metaphase alignment problems, MAP), polylobed nuclei\n",
      "Increased mitotic delay (prometaphase or metaphase alignment problems, MAP), polylobed nuclei, decreased viability (cell death)\n",
      "Increased mitotic delay (prometaphase or metaphase alignment problems, MAP), polylobed nuclei, dynamic nuclei (hole, folded or small irregular)\n",
      "Increased mitotic delay (prometaphase or metaphase alignment problems, MAP), polylobed nuclei, dynamic nuclei (hole, folded or small irregular), decreased viability (cell death)\n",
      "Increased mitotic delay (prometaphase or metaphase alignment problems, MAP), polylobed nuclei, large nuclei\n",
      "Increased mitotic delay (prometaphase or metaphase alignment problems, MAP), polylobed nuclei, many micronuclei (grape)\n",
      "Increased mitotic delay (prometaphase or metaphase alignment problems, MAP), polylobed nuclei, many micronuclei (grape), decreased viability (cell death)\n",
      "Increased mitotic delay (prometaphase or metaphase alignment problems, MAP), polylobed nuclei, many micronuclei (grape), dynamic nuclei (hole, folded or small irregular), decreased viability (cell death)\n",
      "Increased mitotic delay (prometaphase or metaphase alignment problems, MAP), polylobed nuclei, many micronuclei (grape), large nuclei\n",
      "Increased mitotic delay (prometaphase or metaphase alignment problems, MAP), polylobed nuclei, many micronuclei (grape), large nuclei, decreased viability (cell death)\n",
      "Increased mitotic delay (prometaphase or metaphase alignment problems, MAP), two nuclei (binuclear)\n",
      "Increased mitotic delay (prometaphase or metaphase alignment problems, MAP), two nuclei (binuclear), many micronuclei (grape)\n",
      "Increased mitotic delay (prometaphase or metaphase alignment problems, MAP), two nuclei (binuclear), polylobed nuclei\n",
      "Increased mitotic delay (prometaphase or metaphase alignment problems, MAP), two nuclei (binuclear), polylobed nuclei, many micronuclei (grape), dynamic nuclei (hole, folded or small irregular)\n",
      "Increased nuclear migration distance\n",
      "Increased nuclear migration distance, increased nuclear migration speed\n",
      "Increased nuclear migration speed\n",
      "Increased viability (proliferation)\n",
      "Increased viability (proliferation), increased nuclear migration speed\n",
      "Large nuclei, decreased viability (cell death)\n",
      "Large nuclei, dynamic nuclei (hole, folded or small irregular)\n",
      "Large nuclei, dynamic nuclei (hole, folded or small irregular), decreased viability (cell death)\n",
      "Large nuclei, increased nuclear migration distance\n",
      "Large nuclei, increased nuclear migration distance, increased nuclear migration speed\n",
      "Many micronuclei (grape)\n",
      "Many micronuclei (grape), decreased viability (cell death)\n",
      "Many micronuclei (grape), dynamic nuclei (hole, folded or small irregular)\n",
      "Many micronuclei (grape), dynamic nuclei (hole, folded or small irregular), decreased viability (cell death)\n",
      "Many micronuclei (grape), large nuclei\n",
      "Many micronuclei (grape), large nuclei, decreased viability (cell death)\n",
      "Effect on mitosis\n",
      "Effect on mitosis, decreased viability (cell death)\n",
      "Effect on mitosis, dynamic nuclei (hole, folded or small irregular)\n",
      "Effect on mitosis, increased nuclear migration speed\n",
      "Effect on mitosis, increased viability (proliferation)\n",
      "Effect on mitosis, large nuclei\n",
      "Polylobed nuclei\n",
      "Polylobed nuclei, decreased viability (cell death)\n",
      "Polylobed nuclei, dynamic nuclei (hole, folded or small irregular)\n",
      "Polylobed nuclei, dynamic nuclei (hole, folded or small irregular), decreased viability (cell death)\n",
      "Polylobed nuclei, increased nuclear migration speed\n",
      "Polylobed nuclei, large nuclei\n",
      "Polylobed nuclei, large nuclei, decreased viability (cell death)\n",
      "Polylobed nuclei, large nuclei, dynamic nuclei (hole, folded or small irregular)\n",
      "Polylobed nuclei, large nuclei, dynamic nuclei (hole, folded or small irregular), decreased viability (cell death)\n",
      "Polylobed nuclei, many micronuclei (grape)\n",
      "Polylobed nuclei, many micronuclei (grape), decreased viability (cell death)\n",
      "Polylobed nuclei, many micronuclei (grape), dynamic nuclei (hole, folded or small irregular)\n",
      "Polylobed nuclei, many micronuclei (grape), dynamic nuclei (hole, folded or small irregular), decreased viability (cell death)\n",
      "Polylobed nuclei, many micronuclei (grape), large nuclei\n",
      "Polylobed nuclei, many micronuclei (grape), large nuclei, decreased viability (cell death)\n",
      "Two nuclei (binuclear)\n",
      "Two nuclei (binuclear), decreased viability (cell death)\n",
      "Two nuclei (binuclear), dynamic nuclei (hole, folded or small irregular)\n",
      "Two nuclei (binuclear), dynamic nuclei (hole, folded or small irregular), decreased viability (cell death)\n",
      "Two nuclei (binuclear), increased nuclear migration distance, increased nuclear migration speed\n",
      "Two nuclei (binuclear), increased nuclear migration speed\n",
      "Two nuclei (binuclear), large nuclei\n",
      "Two nuclei (binuclear), many micronuclei (grape)\n",
      "Two nuclei (binuclear), many micronuclei (grape), decreased viability (cell death)\n",
      "Two nuclei (binuclear), polylobed nuclei\n",
      "Two nuclei (binuclear), polylobed nuclei, decreased viability (cell death)\n",
      "Two nuclei (binuclear), polylobed nuclei, dynamic nuclei (hole, folded or small irregular)\n",
      "Two nuclei (binuclear), polylobed nuclei, dynamic nuclei (hole, folded or small irregular), decreased viability (cell death)\n",
      "Two nuclei (binuclear), polylobed nuclei, large nuclei, dynamic nuclei (hole, folded or small irregular)\n",
      "Two nuclei (binuclear), polylobed nuclei, many micronuclei (grape)\n",
      "Two nuclei (binuclear), polylobed nuclei, many micronuclei (grape), dynamic nuclei (hole, folded or small irregular)\n",
      "Decreased viability (cell death), increased nuclear migration distance\n",
      "Decreased viability (cell death), increased nuclear migration distance, increased nuclear migration speed\n",
      "Decreased viability (cell death), increased nuclear migration speed\n",
      "Increased  Cyt C expression\n",
      "Increased centriole number\n",
      "Decreased centriole number\n",
      "Increased shRNA abundance with PLX4720\n",
      "Downregulation of RLR pathway after Sendai virus (SeV) infection\n",
      "Upregulation of RLR pathway after Sendai virus (SeV) infection\n",
      "Decreased Aire reporter expression\n",
      "Decreased human papilloma virus 16 (HPV16) pseudovirus infection\n",
      "Decreased interferon-beta reporter expression\n",
      "Increased interferon-beta reporter expression\n",
      "Decreased Sindbis virus (SINV) infection\n",
      "Increased Sindbis virus (SINV) infection\n",
      "Inconclusive\n",
      "Decreased NF-kappaB reporter expression\n",
      "Increased NF-kappaB reporter expression\n",
      "Downregulation of NFkappaB pathway\n",
      "Decreased caspase 3/7 activity\n",
      "Increased shRNA abundance\n",
      "Decreased LDL uptake\n",
      "Increased LDL uptake\n",
      "Increased free cholesterol\n",
      "Decreased free cholesterol\n",
      "shRNA abundance <= 50%\n",
      "shRNA abundance <= 25%\n",
      "Decreased Dengue Infection\n",
      "Increased Dengue infection\n",
      "Decreased infection with West Nile virus (WNV)\n",
      "Increased infection with West Nile virus (WNV)\n",
      "Increased West Nile virus (WNV) infection\n",
      "Decreased West Nile virus (WNV) infection\n",
      "Decreased HPV16-GFP infection\n",
      "Increased HPV16-GFP infection\n",
      "Resistant to vaccinia virus (VACV-A4L) infection\n",
      "Decreased vaccinia virus (VACV-LREV ) gene expression\n",
      "Decreased cell proliferation\n",
      "Increased transcriptional activity\n",
      "Increased epidermal growth factor receptor (EGFR) surface abundance\n",
      "Decreased epidermal growth factor receptor (EGFR) surface abundance\n",
      "Increased vesicular stomatitis virus (VSV) infection\n",
      "Increased simian virus 40 (SV40) infection\n",
      "Decreased simian virus 40 (SV40) infection\n",
      "Transferrin accumulation in enlarged cytoplasmic structures\n",
      "Transferrin accumulation in small spots in the cell periphery\n",
      "toxic\n",
      "Transferrin accumulation in small spots in the cell periphery, transferrin accumulation in enlarged cytoplasmic structures\n",
      "Transferrin accumulation in small spots in the cell periphery, polarized transferrin accumulation in plasma membrane protrusions\n",
      "Transferrin accumulation in the perinuclear area\n",
      "(Transferrin accumulation in small spots in the cell periphery)\n",
      "(Transferrin accumulation in enlarged cytoplasmic structures)\n",
      "(Polarized transferrin accumulation in plasma membrane protrusions)\n",
      "Polarized transferrin accumulation in plasma membrane protrusions\n",
      "(Transferrin accumulation in the perinuclear area)\n",
      "(Transferrin accumulation in small spots in the cell periphery, polarized transferrin accumulation in plasma membrane protrusions)\n",
      "Transferrin accumulation in enlarged cytoplasmic structures, polarized transferrin accumulation in plasma membrane protrusions\n",
      "Toxic, transferrin accumulation in the perinuclear area\n",
      "Weakly decreased NFAT1-GFP nuclear translocation\n",
      "Moderately decreased NFAT1-GFP nuclear translocation\n",
      "Strongly decreased NFAT1-GFP nuclear translocation\n",
      "Mildly decreased CFP-tsO45G cell surface transport\n",
      "Strongly decreased CFP-tsO45G cell surface transport\n",
      "Increased CFP-tsO45G cell surface transport\n",
      "No cells\n",
      "Decreased replication of vaccinia virus (VACV)\n",
      "Synthetic lethal with vaccinia virus (VACV) infection\n",
      "Increased replication of vaccinia virus (VACV)\n",
      "Decreased transferrin (TF) endocytosis, decreased endosome-nucleus distance (endosomes clustered in the perinuclear region)\n",
      "Increased epidermal growth factor (EGF) endocytosis, decreased endosome-nucleus distance (endosomes clustered in the perinuclear region)\n",
      "Decreased epidermal growth factor (EGF) endocytosis, decreased transferrin (TF) endocytosis\n",
      "Decreased epidermal growth factor (EGF) endocytosis\n",
      "Decreased endosome-nucleus distance (endosomes clustered in the perinuclear region)\n",
      "Decreased transferrin (TF) endocytosis\n",
      "Increased transferrin (TF) endocytosis\n",
      "Decreased epidermal growth factor (EGF) endocytosis, increased transferrin (TF) endocytosis\n",
      "Increased epidermal growth factor (EGF) endocytosis, increased transferrin (TF) endocytosis\n",
      "Increased epidermal growth factor (EGF) endocytosis, decreased transferrin (TF) endocytosis\n",
      "Increased endosome-nucleus distance (endosomes dispersed in the cell periphery)\n",
      "Increased epidermal growth factor (EGF) endocytosis\n",
      "Increased transferrin (TF) endocytosis (mild increase), increased transferrin (TF) endosome elongation\n",
      "Increased epidermal growth factor (EGF) endosome number, increased epidermal growth factor (EGF) endosome elongation\n",
      "Decreased Salmonella proliferation\n",
      "Increased Salmonella proliferation\n",
      "Diffuse trans-Golgi\n",
      "Fragmented cis-Golgi\n",
      "Condensed cis-Golgi, condensed medial Golgi, condensed trans-Golgi\n",
      "Diffuse medial Golgi\n",
      "Fragmented cis-Golgi, fragmented medial Golgi, fragmented trans-Golgi\n",
      "Condensed medial Golgi, diffuse trans-Golgi\n",
      "Diffuse cis-Golgi, diffuse trans-Golgi\n",
      "Condensed medial Golgi\n",
      "Condensed cis-Golgi, condensed medial Golgi\n",
      "Diffuse cis-Golgi\n",
      "Condensed trans-Golgi\n",
      "Fragmented cis-Golgi, fragmented medial Golgi, diffuse trans-Golgi\n",
      "Fragmented cis-Golgi, fragmented trans-Golgi\n",
      "Condensed medial Golgi, condensed trans-Golgi\n",
      "Condensed cis-Golgi\n",
      "Condensed cis-Golgi, condensed trans-Golgi\n",
      "Fragmented trans-Golgi\n",
      "Diffuse cis-Golgi, fragmented trans-Golgi\n",
      "Diffuse cis-Golgi, diffuse medial Golgi\n",
      "Diffuse cis-Golgi, fragmented medial Golgi, fragmented trans-Golgi\n",
      "Fragmented medial Golgi, fragmented trans-Golgi\n",
      "Diffuse cis-Golgi, diffuse medial Golgi, diffuse trans-Golgi\n",
      "Fragmented cis-Golgi, diffuse trans-Golgi\n",
      "Fragmented cis-Golgi, fragmented medial Golgi\n",
      "Increased shRNA abundance (Z-score > 2)\n",
      "Decreased shRNA abundance (Z-score < -2)\n",
      "Decreased viability ratio\n",
      "Reduced shRNA abundance in FBW7 KO line\n",
      "Increased Nanog expression\n",
      "Increased viability with SS1P at EC90\n",
      "Decreased viability with SS1P at EC30\n",
      "Increased virus production\n",
      "Decreased IL-8 secretion\n",
      "Increased IL-8 secretion\n",
      "Increased CDKN1A mRNA expression\n",
      "Decreased CDKN1A mRNA expression\n",
      "Increased BBC3 mRNA expression\n",
      "Decreased BBC3 mRNA expression\n",
      "Decreased TP53 mRNA expression\n",
      "Increased TP53 mRNA expression\n",
      "Reduced mammosphere formation\n",
      "Increased caspase activity\n",
      "no effect\n",
      "Increased the percentage of infected cells\n"
     ]
    }
   ],
   "source": [
    "# Lista con todos los fenotipos individuales de GenomeRNAi.\n",
    "Phenotypes = (DataGRNAi['Phenotype'].astype('category')).unique().tolist()\n",
    "for Phe in Phenotypes:\n",
    "    print(Phe)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Ras\n",
      "cisplatin\n",
      "gemcitabine\n",
      "paclitaxel\n",
      "imatinib mesylate\n",
      "c-Myc after tamoxifen stimulation\n",
      "MLN4924 (a NAE inhibitor)\n",
      "vaccinia virus (VACV) infection\n",
      "['Synthetic lethal with Ras', 'Synthetic lethal with cisplatin', 'Synthetic lethal with gemcitabine', 'Synthetic lethal with paclitaxel', 'Synthetic lethal with imatinib mesylate', 'Synthetic lethal with c-Myc after tamoxifen stimulation', 'Synthetic lethal with MLN4924 (a NAE inhibitor)', 'Synthetic lethal with vaccinia virus (VACV) infection']\n"
     ]
    }
   ],
   "source": [
    "# Recopilación de los fenotipos implicados en algún tipo de interacción de SL.\n",
    "Phenos = []\n",
    "for Phe in Phenotypes:\n",
    "    if (type(Phe) is str) and ('Synthetic lethal' in Phe):\n",
    "        Phenos.append(Phe)\n",
    "        print(Phe[22:])\n",
    "\n",
    "print(Phenos)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     Gene Symbol                                          Phenotype\n",
      "0           UBA2                          Synthetic lethal with Ras\n",
      "1          DIDO1                          Synthetic lethal with Ras\n",
      "2           AMFR                          Synthetic lethal with Ras\n",
      "3         WDR42C                          Synthetic lethal with Ras\n",
      "4         TMEM17                          Synthetic lethal with Ras\n",
      "...          ...                                                ...\n",
      "7249       PLCD1  Synthetic lethal with vaccinia virus (VACV) in...\n",
      "7250        TBCA  Synthetic lethal with vaccinia virus (VACV) in...\n",
      "7251      NHP2L1  Synthetic lethal with vaccinia virus (VACV) in...\n",
      "7252       FOXM1  Synthetic lethal with vaccinia virus (VACV) in...\n",
      "7253      PCOLCE  Synthetic lethal with vaccinia virus (VACV) in...\n",
      "\n",
      "[7254 rows x 2 columns]\n"
     ]
    }
   ],
   "source": [
    "# Filtrado de los datos de GenomeRNAi basado en los fenotipos de SL extraídos previamente.\n",
    "SLGRNAi = DataGRNAi[DataGRNAi['Phenotype'].isin(Phenos)].reset_index(drop = True)\n",
    "print(SLGRNAi)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     n1.name                          n2.name\n",
      "0       UBA2                              Ras\n",
      "1      DIDO1                              Ras\n",
      "2       AMFR                              Ras\n",
      "3     WDR42C                              Ras\n",
      "4     TMEM17                              Ras\n",
      "...      ...                              ...\n",
      "7249   PLCD1  vaccinia virus (VACV) infection\n",
      "7250    TBCA  vaccinia virus (VACV) infection\n",
      "7251  NHP2L1  vaccinia virus (VACV) infection\n",
      "7252   FOXM1  vaccinia virus (VACV) infection\n",
      "7253  PCOLCE  vaccinia virus (VACV) infection\n",
      "\n",
      "[4219 rows x 2 columns]\n"
     ]
    }
   ],
   "source": [
    "SLGRNAi['Phenotype'] = SLGRNAi['Phenotype'].apply(lambda x: x[22:])\n",
    "SLGRNAi = (SLGRNAi.drop_duplicates()).rename(columns = {'Gene Symbol':'n1.name','Phenotype':'n2.name'})\n",
    "print(SLGRNAi)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------SL pairs for Essential PatuT Genes in GenomeRNAi:\n",
      "\n",
      "SL pairs in GenomeRNAi for PRMT3: No SL pairs Found\n",
      "\n",
      "SL pairs in GenomeRNAi for ASXL1: No SL pairs Found\n",
      "\n",
      "SL pairs in GenomeRNAi for HELLS: No SL pairs Found\n",
      "\n",
      "SL pairs in GenomeRNAi for SMARCE1: No SL pairs Found\n",
      "\n",
      "SL pairs in GenomeRNAi for KDM4B: No SL pairs Found\n",
      "\n",
      "SL pairs in GenomeRNAi for EPC2: No SL pairs Found\n",
      "\n",
      "-------SL pairs for Suppressor PatuT Genes in GenomeRNAi:\n",
      "\n",
      "SL pairs in GenomeRNAi for WHSC1: No SL pairs Found\n",
      "\n",
      "SL pairs in GenomeRNAi for SUV39H2: ['Ras']\n",
      "\n",
      "-------SL pairs for Essential PatuS Genes in GenomeRNAi:\n",
      "\n",
      "SL pairs in GenomeRNAi for CHD7: ['Ras']\n",
      "\n",
      "SL pairs in GenomeRNAi for BRD4: ['c-Myc after tamoxifen stimulation', 'MLN4924 (a NAE inhibitor)']\n",
      "\n",
      "SL pairs in GenomeRNAi for ING5: No SL pairs Found\n",
      "\n",
      "-------SL pairs for Suppressor PatuS Genes in GenomeRNAi:\n",
      "\n",
      "SL pairs in GenomeRNAi for PHF23: No SL pairs Found\n",
      "\n",
      "SL pairs in GenomeRNAi for UBE2A: ['cisplatin', 'MLN4924 (a NAE inhibitor)']\n",
      "\n"
     ]
    }
   ],
   "source": [
    "SLcsv(SLGRNAi,'GenomeRNAi')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Gold Standard"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "        gene1       gene2 gene1 perturbation gene2 perturbation      PMID  \\\n",
      "0       MUS81       TTC31                mut              shRNA  24104479   \n",
      "1       MUS81       ESPL1                mut              shRNA  24104479   \n",
      "2       MUS81  ST6GALNAC3                mut              shRNA  24104479   \n",
      "3       MUS81       MTMR6                mut              shRNA  24104479   \n",
      "4       MUS81       OVCH1                mut              shRNA  24104479   \n",
      "...       ...         ...                ...                ...       ...   \n",
      "154702   ALAD      NDUFB1              sgRNA              sgRNA  28319085   \n",
      "154703  MST1R      PFKFB4              sgRNA              sgRNA  28319085   \n",
      "154704  CNTFR        NPR2              sgRNA              sgRNA  28319085   \n",
      "154705    BSG         GPI              sgRNA              sgRNA  28319085   \n",
      "154706    CA9        NPR2              sgRNA              sgRNA  28319085   \n",
      "\n",
      "       cancer type tested  SL  \n",
      "0                    COAD   0  \n",
      "1                    COAD   1  \n",
      "2                    COAD   0  \n",
      "3                    COAD   0  \n",
      "4                    COAD   0  \n",
      "...                   ...  ..  \n",
      "154702               LAML   0  \n",
      "154703               LAML   0  \n",
      "154704               LAML   0  \n",
      "154705               LAML   0  \n",
      "154706               LAML   0  \n",
      "\n",
      "[154707 rows x 7 columns]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_17508/3506160960.py:1: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  SLgs = pd.read_csv(f'{work_folder}Gold_Standard/41467_2018_4647_MOESM4_ESM.csv')\n"
     ]
    }
   ],
   "source": [
    "# Carga de los datos del GS:\n",
    "SLgs = pd.read_csv(f'{work_folder}Gold_Standard/41467_2018_4647_MOESM4_ESM.csv')\n",
    "print(SLgs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       n1.name  n2.name\n",
      "1        MUS81    ESPL1\n",
      "6        MUS81    TRAF6\n",
      "9        MUS81    TRIP6\n",
      "17       MUS81  PLEKHH1\n",
      "20       MUS81   CREBBP\n",
      "...        ...      ...\n",
      "135394  CAMK2A      OXT\n",
      "135395   BIRC5    MMP14\n",
      "135396   KCNQ4    KDM1A\n",
      "135397  CAMK1G   PIK3CA\n",
      "135398  PIK3R2    STMN1\n",
      "\n",
      "[6033 rows x 2 columns]\n"
     ]
    }
   ],
   "source": [
    "SLgs = ((SLgs[SLgs['SL'] == 1])[['gene1', 'gene2']]).rename(columns = {'gene1':'n1.name','gene2':'n2.name'})\n",
    "print(SLgs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------SL pairs for Essential PatuT Genes in Gold_Standard:\n",
      "\n",
      "SL pairs in Gold_Standard for PRMT3: No SL pairs Found\n",
      "\n",
      "SL pairs in Gold_Standard for ASXL1: ['MUS81']\n",
      "\n",
      "SL pairs in Gold_Standard for HELLS: No SL pairs Found\n",
      "\n",
      "SL pairs in Gold_Standard for SMARCE1: No SL pairs Found\n",
      "\n",
      "SL pairs in Gold_Standard for KDM4B: No SL pairs Found\n",
      "\n",
      "SL pairs in Gold_Standard for EPC2: No SL pairs Found\n",
      "\n",
      "-------SL pairs for Suppressor PatuT Genes in Gold_Standard:\n",
      "\n",
      "SL pairs in Gold_Standard for WHSC1: No SL pairs Found\n",
      "\n",
      "SL pairs in Gold_Standard for SUV39H2: ['KRAS']\n",
      "\n",
      "-------SL pairs for Essential PatuS Genes in Gold_Standard:\n",
      "\n",
      "SL pairs in Gold_Standard for CHD7: No SL pairs Found\n",
      "\n",
      "SL pairs in Gold_Standard for BRD4: ['MYC', 'MAP2K1', 'DHFR', 'FGFR3', 'KDM5C', 'PTEN', 'RB1', 'PTAR1', 'WEE1', 'CDC7', 'CDK9', 'CHEK1', 'CHEK2', 'WEE1', 'CDC7', 'CDK9', 'CHEK1', 'CHEK2', 'BRCA1']\n",
      "\n",
      "SL pairs in Gold_Standard for ING5: ['CDC7', 'CDK9', 'MAP2K1', 'TUBA1A', 'CDC7', 'CDK9', 'MAP2K1', 'TUBA1A']\n",
      "\n",
      "-------SL pairs for Suppressor PatuS Genes in Gold_Standard:\n",
      "\n",
      "SL pairs in Gold_Standard for PHF23: No SL pairs Found\n",
      "\n",
      "SL pairs in Gold_Standard for UBE2A: No SL pairs Found\n",
      "\n"
     ]
    }
   ],
   "source": [
    "SLcsv(SLgs,'Gold_Standard')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
